{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pylab as plt\n",
    "import seaborn as sns\n",
    "import os\n",
    "import gzip\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# chrom_bp_net # annovar_data # allele frequency"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<function gzip.open(filename, mode='rb', compresslevel=9, encoding=None, errors=None, newline=None)>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gzip.open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['allele_freq_chr19_gene_ncr.tsv', 'allele_freq_chr12_gene_ncr.tsv', 'allele_freq_chr6_gene_ncr.tsv', 'allele_freq_chr1_gene_ncr.tsv', 'allele_freq_chr3_gene_ncr.tsv', 'allele_freq_chr7_gene_ncr.tsv', 'allele_freq_chr16_gene_ncr.tsv', 'allele_freq_chr17_gene_ncr.tsv', 'allele_freq_chr13_gene_ncr.tsv', 'allele_freq_chr11_gene_ncr.tsv', 'allele_freq_chr15_gene_ncr.tsv', 'allele_freq_chr9_gene_ncr.tsv', 'allele_freq_chr5_gene_ncr.tsv', 'allele_freq_chr10_gene_ncr.tsv', 'allele_freq_chr8_gene_ncr.tsv', 'allele_freq_chr21_gene_ncr.tsv', 'allele_freq_chr14_gene_ncr.tsv', 'allele_freq_chr2_gene_ncr.tsv', 'allele_freq_chr18_gene_ncr.tsv', 'allele_freq_chr22_gene_ncr.tsv', 'allele_freq_chr4_gene_ncr.tsv', 'allele_freq_chr20_gene_ncr.tsv']\n"
     ]
    }
   ],
   "source": [
    "directory = '/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model4'\n",
    "directory1 = '/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model4/annovar'\n",
    "# directory2 = '/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model4/annovar2'\n",
    "\n",
    "\n",
    "filename_af1=[]\n",
    "filename_co1=[]\n",
    "filename1=[]\n",
    "filename_anno1=[]\n",
    "\n",
    "\n",
    "for filename_af in os.listdir(directory):\n",
    "    if filename_af.endswith('gene_ncr.tsv'):\n",
    "        with open(os.path.join(directory, filename_af)) as f:\n",
    "            filename_af1.append(filename_af)\n",
    "print(filename_af1)\n",
    "            \n",
    "        \n",
    "            \n",
    "for filename_co in os.listdir(directory):\n",
    "    if filename_co.startswith('chrmbpnet_out'):\n",
    "        with open(os.path.join(directory, filename_co)) as f:\n",
    "            filename_co1.append(filename_co)\n",
    "            \n",
    "            \n",
    "            #print(filename_co)\n",
    "\n",
    "for filename in os.listdir(directory1):\n",
    "    if (filename.endswith('txt.gz') and filename.startswith('anovar_gene')):\n",
    "        with gzip.open(os.path.join(directory1, filename)) as f:\n",
    "            #print(filename)\n",
    "            filename1.append(filename)\n",
    "            \n",
    "\n",
    "# for filename_anno in os.listdir(directory2):\n",
    "#     if (filename_anno.endswith('gene.hg38_wgEncodeBroadHmmGm12878HMM')):\n",
    "#         with gzip.open(os.path.join(directory2, filename_anno)) as f:\n",
    "#             #print(filename)\n",
    "#             filename_anno1.append(filename_anno)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filename_anno1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define a sorting key function to extract the integer from the file name\n",
    "def extract_integer(file_name):\n",
    "    return int(re.search(r'\\d+', file_name).group())\n",
    "\n",
    "# Sort the file names based on the extracted integer\n",
    "sorted_filename_af1= sorted(filename_af1, key=extract_integer)\n",
    "sorted_filename_co1= sorted(filename_co1, key=extract_integer)\n",
    "sorted_filename1= sorted(filename1, key=extract_integer)\n",
    "# sorted_filename_anno1= sorted(filename_anno1, key=extract_integer)\n",
    "\n",
    "# # Loop over the sorted file names\n",
    "# for i,j,k in zip(sorted_filename_af1,sorted_filename_co1,sorted_filename1):\n",
    "#     # Process each file in the loop\n",
    "#     print(i,j,k)\n",
    "#     # Add your code here to perform operations on each file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['anovar_gene_outliers_chr1.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr2.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr3.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr4.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr5.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr6.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr7.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr8.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr9.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr10.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr11.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr12.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr13.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr14.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr15.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr16.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr17.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr18.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr19.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr20.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr21.hg38_multianno.txt.gz',\n",
       " 'anovar_gene_outliers_chr22.hg38_multianno.txt.gz']"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted_filename1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "anovar_gene_outliers_chr1.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr2.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr3.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr4.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr5.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr6.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr7.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr8.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr9.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr10.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr11.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr12.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr13.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr14.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr15.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr16.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr17.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr18.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr19.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr20.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr21.hg38_multianno.txt.gz\n",
      "anovar_gene_outliers_chr22.hg38_multianno.txt.gz\n"
     ]
    }
   ],
   "source": [
    "# Loop over the sorted file names\n",
    "l=0\n",
    "for i,j,k in zip(sorted_filename_af1,sorted_filename_co1,sorted_filename1):\n",
    "    print(k)\n",
    "    full_paths_af = os.path.join(directory, i)\n",
    "    full_paths_co = os.path.join(directory, j)\n",
    "    full_paths_an = os.path.join(directory1, k)\n",
    "#     full_paths_anno = os.path.join(directory2, m)\n",
    "\n",
    "    \n",
    "    data_af=pd.read_csv(full_paths_af,sep='\\t',skiprows=[0],header=None,names=[\"chr\",\"pos\",\"ref\",\"alt\",\"allele_freq\"])   \n",
    "    data_co=pd.read_csv(full_paths_co,sep='\\t',names=[\"chr\",\"pos\",\"ref\",\"alt\",\"log_counts_diff\",\"log_probs_diff_abs_sum\",\"probs_jsd_diff\"])\n",
    "    data_an1=pd.read_csv(full_paths_an,sep='\\t', usecols=range(24))\n",
    "#     data_anno=pd.read_csv(full_paths_anno,sep='\\t',header=None,usecols=[1,2,3],names=[\"anno\",\"chr\",\"pos\"])\n",
    "\n",
    "    \n",
    "    \n",
    "    data_an=data_an1.rename(columns={\"Start\": \"pos\"})\n",
    "    \n",
    "    #data_af['pos'] = data_af['pos'].astype(int)\n",
    "    #data_co['pos'] = data_co['pos'].astype(int)\n",
    "\n",
    "    #data_anno['pos'] = data_anno['pos'].astype(int)\n",
    "\n",
    "    \n",
    "    merged_df = pd.merge(data_af, data_co, on='pos', how='left')\n",
    "    merged_df = pd.merge(merged_df, data_an, on='pos', how='left')\n",
    "    #merged_df = pd.merge(merged_df, data_anno, on='pos', how='inner')\n",
    "    \n",
    "    \n",
    "    df=merged_df.drop(columns=[\"chr_y\",\"ref_y\",\"alt_y\",\"Chr\",\"End\",\"Ref\",\"Alt\"])\n",
    "    df.to_csv(os.path.join(directory, \"anno_af_chrom_new/\"+str(i)+\"_chr_\"+str(l+1)+'.csv'),index=None)\n",
    "\n",
    "    #df.to_csv(\"/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model1/anno_af_chrom/\"+str(i)+\"_chr_\"+str(l+1)+'.csv',index=None)\n",
    "    \n",
    "    \n",
    "    l=l+1\n",
    "#     if l==1:\n",
    "#         break\n",
    "    \n",
    "\n",
    "        # Process each file in the loop\n",
    "    #print(i,j,k)\n",
    "    # Add your code here to perform operations on each file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'data_anno' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-27-49797ffdf8a1>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdata_anno\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'data_anno' is not defined"
     ]
    }
   ],
   "source": [
    "data_anno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(data_co.dtypes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    " data_co"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr_x</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad40_genome_AF_ami</th>\n",
       "      <th>gnomad40_genome_AF_amr</th>\n",
       "      <th>gnomad40_genome_AF_asj</th>\n",
       "      <th>gnomad40_genome_AF_eas</th>\n",
       "      <th>gnomad40_genome_AF_fin</th>\n",
       "      <th>gnomad40_genome_AF_mid</th>\n",
       "      <th>gnomad40_genome_AF_nfe</th>\n",
       "      <th>gnomad40_genome_AF_remaining</th>\n",
       "      <th>gnomad40_genome_AF_sas</th>\n",
       "      <th>anno</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr22</td>\n",
       "      <td>22094965</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.027652</td>\n",
       "      <td>0.016994</td>\n",
       "      <td>87.278244</td>\n",
       "      <td>0.041332</td>\n",
       "      <td>0.0858</td>\n",
       "      <td>0.0862</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0582</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.0825</td>\n",
       "      <td>0.0416</td>\n",
       "      <td>0.0204</td>\n",
       "      <td>0.0100</td>\n",
       "      <td>0.0592</td>\n",
       "      <td>0.0170</td>\n",
       "      <td>Name=11_Weak_Txn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr22</td>\n",
       "      <td>22105020</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.418073</td>\n",
       "      <td>0.014939</td>\n",
       "      <td>167.423250</td>\n",
       "      <td>0.098816</td>\n",
       "      <td>0.3817</td>\n",
       "      <td>0.3820</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2182</td>\n",
       "      <td>0.3338</td>\n",
       "      <td>0.2778</td>\n",
       "      <td>0.6203</td>\n",
       "      <td>0.2227</td>\n",
       "      <td>0.2789</td>\n",
       "      <td>0.2198</td>\n",
       "      <td>0.3581</td>\n",
       "      <td>0.3486</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr22</td>\n",
       "      <td>22112777</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.277391</td>\n",
       "      <td>0.016652</td>\n",
       "      <td>98.462780</td>\n",
       "      <td>0.039174</td>\n",
       "      <td>0.0949</td>\n",
       "      <td>0.0951</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0669</td>\n",
       "      <td>0.0676</td>\n",
       "      <td>0.0902</td>\n",
       "      <td>0.4773</td>\n",
       "      <td>0.0914</td>\n",
       "      <td>0.0952</td>\n",
       "      <td>0.0532</td>\n",
       "      <td>0.0942</td>\n",
       "      <td>0.2200</td>\n",
       "      <td>Name=5_Strong_Enhancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr22</td>\n",
       "      <td>22468026</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.170087</td>\n",
       "      <td>0.018848</td>\n",
       "      <td>109.472140</td>\n",
       "      <td>0.049200</td>\n",
       "      <td>0.1174</td>\n",
       "      <td>0.1178</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0614</td>\n",
       "      <td>0.1110</td>\n",
       "      <td>0.0815</td>\n",
       "      <td>0.1542</td>\n",
       "      <td>0.0953</td>\n",
       "      <td>0.1748</td>\n",
       "      <td>0.0724</td>\n",
       "      <td>0.1162</td>\n",
       "      <td>0.1741</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr22</td>\n",
       "      <td>22581344</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.443652</td>\n",
       "      <td>0.015209</td>\n",
       "      <td>100.493210</td>\n",
       "      <td>0.045920</td>\n",
       "      <td>0.3241</td>\n",
       "      <td>0.3259</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2923</td>\n",
       "      <td>0.4322</td>\n",
       "      <td>0.2412</td>\n",
       "      <td>0.7667</td>\n",
       "      <td>0.2028</td>\n",
       "      <td>0.3462</td>\n",
       "      <td>0.2119</td>\n",
       "      <td>0.3483</td>\n",
       "      <td>0.4103</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr22</td>\n",
       "      <td>22586441</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.668812</td>\n",
       "      <td>0.021155</td>\n",
       "      <td>112.381004</td>\n",
       "      <td>0.052819</td>\n",
       "      <td>0.6092</td>\n",
       "      <td>0.6097</td>\n",
       "      <td>...</td>\n",
       "      <td>0.6206</td>\n",
       "      <td>0.6896</td>\n",
       "      <td>0.6040</td>\n",
       "      <td>0.7428</td>\n",
       "      <td>0.6284</td>\n",
       "      <td>0.5951</td>\n",
       "      <td>0.5841</td>\n",
       "      <td>0.6357</td>\n",
       "      <td>0.6683</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr22</td>\n",
       "      <td>25269888</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.451551</td>\n",
       "      <td>0.016266</td>\n",
       "      <td>84.820366</td>\n",
       "      <td>0.034757</td>\n",
       "      <td>0.5258</td>\n",
       "      <td>0.5571</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4962</td>\n",
       "      <td>0.6245</td>\n",
       "      <td>0.4821</td>\n",
       "      <td>0.5804</td>\n",
       "      <td>0.4855</td>\n",
       "      <td>0.5602</td>\n",
       "      <td>0.5225</td>\n",
       "      <td>0.5178</td>\n",
       "      <td>0.4364</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr22</td>\n",
       "      <td>25269896</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.453881</td>\n",
       "      <td>0.017258</td>\n",
       "      <td>140.720460</td>\n",
       "      <td>0.069248</td>\n",
       "      <td>0.5235</td>\n",
       "      <td>0.5606</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4895</td>\n",
       "      <td>0.6194</td>\n",
       "      <td>0.4763</td>\n",
       "      <td>0.5740</td>\n",
       "      <td>0.4785</td>\n",
       "      <td>0.5639</td>\n",
       "      <td>0.5173</td>\n",
       "      <td>0.5159</td>\n",
       "      <td>0.4282</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr22</td>\n",
       "      <td>25273952</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.137472</td>\n",
       "      <td>0.016103</td>\n",
       "      <td>128.067500</td>\n",
       "      <td>0.075961</td>\n",
       "      <td>0.2565</td>\n",
       "      <td>0.1950</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2222</td>\n",
       "      <td>0.2054</td>\n",
       "      <td>0.2896</td>\n",
       "      <td>0.0632</td>\n",
       "      <td>0.1215</td>\n",
       "      <td>0.2478</td>\n",
       "      <td>0.1878</td>\n",
       "      <td>0.2453</td>\n",
       "      <td>0.2055</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr22</td>\n",
       "      <td>26831256</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.863565</td>\n",
       "      <td>0.015903</td>\n",
       "      <td>130.792570</td>\n",
       "      <td>0.068628</td>\n",
       "      <td>0.7391</td>\n",
       "      <td>0.7390</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8026</td>\n",
       "      <td>0.7515</td>\n",
       "      <td>0.5765</td>\n",
       "      <td>0.9641</td>\n",
       "      <td>0.7572</td>\n",
       "      <td>0.7279</td>\n",
       "      <td>0.6921</td>\n",
       "      <td>0.7277</td>\n",
       "      <td>0.8335</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr22</td>\n",
       "      <td>27034195</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.323652</td>\n",
       "      <td>0.014330</td>\n",
       "      <td>100.264370</td>\n",
       "      <td>0.054446</td>\n",
       "      <td>0.3021</td>\n",
       "      <td>0.3024</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2004</td>\n",
       "      <td>0.2715</td>\n",
       "      <td>0.2656</td>\n",
       "      <td>0.2284</td>\n",
       "      <td>0.2924</td>\n",
       "      <td>0.2245</td>\n",
       "      <td>0.2733</td>\n",
       "      <td>0.2950</td>\n",
       "      <td>0.3244</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr22</td>\n",
       "      <td>27074801</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.974870</td>\n",
       "      <td>0.014776</td>\n",
       "      <td>103.785260</td>\n",
       "      <td>0.063739</td>\n",
       "      <td>0.8853</td>\n",
       "      <td>0.8853</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9803</td>\n",
       "      <td>0.9080</td>\n",
       "      <td>0.9052</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>0.9072</td>\n",
       "      <td>0.9728</td>\n",
       "      <td>0.8647</td>\n",
       "      <td>0.9001</td>\n",
       "      <td>0.9679</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr22</td>\n",
       "      <td>27263668</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.692174</td>\n",
       "      <td>0.015610</td>\n",
       "      <td>79.606230</td>\n",
       "      <td>0.025526</td>\n",
       "      <td>0.6037</td>\n",
       "      <td>0.6049</td>\n",
       "      <td>...</td>\n",
       "      <td>0.7826</td>\n",
       "      <td>0.6405</td>\n",
       "      <td>0.6409</td>\n",
       "      <td>0.6550</td>\n",
       "      <td>0.6208</td>\n",
       "      <td>0.6103</td>\n",
       "      <td>0.6411</td>\n",
       "      <td>0.6187</td>\n",
       "      <td>0.6790</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22</td>\n",
       "      <td>27478426</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.289826</td>\n",
       "      <td>0.014661</td>\n",
       "      <td>69.238590</td>\n",
       "      <td>0.029925</td>\n",
       "      <td>0.2582</td>\n",
       "      <td>0.2678</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4973</td>\n",
       "      <td>0.2197</td>\n",
       "      <td>0.3844</td>\n",
       "      <td>0.1176</td>\n",
       "      <td>0.3131</td>\n",
       "      <td>0.2375</td>\n",
       "      <td>0.3585</td>\n",
       "      <td>0.2695</td>\n",
       "      <td>0.2663</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr22</td>\n",
       "      <td>27666880</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.999478</td>\n",
       "      <td>0.014991</td>\n",
       "      <td>100.585860</td>\n",
       "      <td>0.042600</td>\n",
       "      <td>0.9372</td>\n",
       "      <td>0.9373</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9785</td>\n",
       "      <td>0.9997</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9726</td>\n",
       "      <td>0.9991</td>\n",
       "      <td>0.9592</td>\n",
       "      <td>0.9992</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr22</td>\n",
       "      <td>29845072</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.251652</td>\n",
       "      <td>0.015735</td>\n",
       "      <td>60.849297</td>\n",
       "      <td>0.018526</td>\n",
       "      <td>0.1148</td>\n",
       "      <td>0.1150</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0296</td>\n",
       "      <td>0.1500</td>\n",
       "      <td>0.0300</td>\n",
       "      <td>0.1072</td>\n",
       "      <td>0.0296</td>\n",
       "      <td>0.0714</td>\n",
       "      <td>0.0253</td>\n",
       "      <td>0.0965</td>\n",
       "      <td>0.2432</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr22</td>\n",
       "      <td>33940618</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.334870</td>\n",
       "      <td>0.015211</td>\n",
       "      <td>96.350120</td>\n",
       "      <td>0.051596</td>\n",
       "      <td>0.2890</td>\n",
       "      <td>0.2891</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5110</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.3041</td>\n",
       "      <td>0.4909</td>\n",
       "      <td>0.3275</td>\n",
       "      <td>0.3095</td>\n",
       "      <td>0.3553</td>\n",
       "      <td>0.2964</td>\n",
       "      <td>0.3394</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr22</td>\n",
       "      <td>34445017</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.074348</td>\n",
       "      <td>0.015491</td>\n",
       "      <td>70.150510</td>\n",
       "      <td>0.030042</td>\n",
       "      <td>0.1175</td>\n",
       "      <td>0.1179</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0099</td>\n",
       "      <td>0.2025</td>\n",
       "      <td>0.0582</td>\n",
       "      <td>0.2114</td>\n",
       "      <td>0.0157</td>\n",
       "      <td>0.0782</td>\n",
       "      <td>0.0280</td>\n",
       "      <td>0.1117</td>\n",
       "      <td>0.0729</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr22</td>\n",
       "      <td>35697829</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.673074</td>\n",
       "      <td>0.018672</td>\n",
       "      <td>99.254900</td>\n",
       "      <td>0.040608</td>\n",
       "      <td>0.6086</td>\n",
       "      <td>0.6162</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5959</td>\n",
       "      <td>0.5326</td>\n",
       "      <td>0.5555</td>\n",
       "      <td>0.7077</td>\n",
       "      <td>0.5420</td>\n",
       "      <td>0.6016</td>\n",
       "      <td>0.5871</td>\n",
       "      <td>0.6202</td>\n",
       "      <td>0.6602</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr22</td>\n",
       "      <td>36981053</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.524261</td>\n",
       "      <td>0.015764</td>\n",
       "      <td>62.000520</td>\n",
       "      <td>0.028188</td>\n",
       "      <td>0.4840</td>\n",
       "      <td>0.4842</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5454</td>\n",
       "      <td>0.5401</td>\n",
       "      <td>0.5972</td>\n",
       "      <td>0.6021</td>\n",
       "      <td>0.4084</td>\n",
       "      <td>0.5612</td>\n",
       "      <td>0.4799</td>\n",
       "      <td>0.5266</td>\n",
       "      <td>0.5027</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr22</td>\n",
       "      <td>47292477</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.628348</td>\n",
       "      <td>0.015340</td>\n",
       "      <td>65.705310</td>\n",
       "      <td>0.028613</td>\n",
       "      <td>0.6796</td>\n",
       "      <td>0.6796</td>\n",
       "      <td>...</td>\n",
       "      <td>0.6897</td>\n",
       "      <td>0.5682</td>\n",
       "      <td>0.7046</td>\n",
       "      <td>0.5765</td>\n",
       "      <td>0.7013</td>\n",
       "      <td>0.7075</td>\n",
       "      <td>0.6644</td>\n",
       "      <td>0.6430</td>\n",
       "      <td>0.6366</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr22</td>\n",
       "      <td>49219057</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.805130</td>\n",
       "      <td>0.018138</td>\n",
       "      <td>104.419480</td>\n",
       "      <td>0.055738</td>\n",
       "      <td>0.8730</td>\n",
       "      <td>0.8729</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9211</td>\n",
       "      <td>0.8947</td>\n",
       "      <td>0.8616</td>\n",
       "      <td>0.9013</td>\n",
       "      <td>0.8883</td>\n",
       "      <td>0.8571</td>\n",
       "      <td>0.8847</td>\n",
       "      <td>0.8646</td>\n",
       "      <td>0.7897</td>\n",
       "      <td>Name=13_Heterochrom/lo</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr_x       pos ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0   chr22  22094965     T     C     0.027652         0.016994   \n",
       "1   chr22  22105020     T     C     0.418073         0.014939   \n",
       "2   chr22  22112777     T     C     0.277391         0.016652   \n",
       "3   chr22  22468026     T     C     0.170087         0.018848   \n",
       "4   chr22  22581344     A     C     0.443652         0.015209   \n",
       "5   chr22  22586441     T     C     0.668812         0.021155   \n",
       "6   chr22  25269888     A     C     0.451551         0.016266   \n",
       "7   chr22  25269896     T     C     0.453881         0.017258   \n",
       "8   chr22  25273952     T     C     0.137472         0.016103   \n",
       "9   chr22  26831256     T     C     0.863565         0.015903   \n",
       "10  chr22  27034195     A     G     0.323652         0.014330   \n",
       "11  chr22  27074801     T     C     0.974870         0.014776   \n",
       "12  chr22  27263668     A     C     0.692174         0.015610   \n",
       "13  chr22  27478426     T     G     0.289826         0.014661   \n",
       "14  chr22  27666880     A     C     0.999478         0.014991   \n",
       "15  chr22  29845072     A     G     0.251652         0.015735   \n",
       "16  chr22  33940618     A     G     0.334870         0.015211   \n",
       "17  chr22  34445017     T     C     0.074348         0.015491   \n",
       "18  chr22  35697829     T     G     0.673074         0.018672   \n",
       "19  chr22  36981053     T     C     0.524261         0.015764   \n",
       "20  chr22  47292477     T     C     0.628348         0.015340   \n",
       "21  chr22  49219057     T     C     0.805130         0.018138   \n",
       "\n",
       "    log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                87.278244        0.041332             0.0858   \n",
       "1               167.423250        0.098816             0.3817   \n",
       "2                98.462780        0.039174             0.0949   \n",
       "3               109.472140        0.049200             0.1174   \n",
       "4               100.493210        0.045920             0.3241   \n",
       "5               112.381004        0.052819             0.6092   \n",
       "6                84.820366        0.034757             0.5258   \n",
       "7               140.720460        0.069248             0.5235   \n",
       "8               128.067500        0.075961             0.2565   \n",
       "9               130.792570        0.068628             0.7391   \n",
       "10              100.264370        0.054446             0.3021   \n",
       "11              103.785260        0.063739             0.8853   \n",
       "12               79.606230        0.025526             0.6037   \n",
       "13               69.238590        0.029925             0.2582   \n",
       "14              100.585860        0.042600             0.9372   \n",
       "15               60.849297        0.018526             0.1148   \n",
       "16               96.350120        0.051596             0.2890   \n",
       "17               70.150510        0.030042             0.1175   \n",
       "18               99.254900        0.040608             0.6086   \n",
       "19               62.000520        0.028188             0.4840   \n",
       "20               65.705310        0.028613             0.6796   \n",
       "21              104.419480        0.055738             0.8730   \n",
       "\n",
       "   gnomad40_genome_AF_raw  ... gnomad40_genome_AF_ami gnomad40_genome_AF_amr  \\\n",
       "0                  0.0862  ...                 0.0066                 0.0582   \n",
       "1                  0.3820  ...                 0.2182                 0.3338   \n",
       "2                  0.0951  ...                 0.0669                 0.0676   \n",
       "3                  0.1178  ...                 0.0614                 0.1110   \n",
       "4                  0.3259  ...                 0.2923                 0.4322   \n",
       "5                  0.6097  ...                 0.6206                 0.6896   \n",
       "6                  0.5571  ...                 0.4962                 0.6245   \n",
       "7                  0.5606  ...                 0.4895                 0.6194   \n",
       "8                  0.1950  ...                 0.2222                 0.2054   \n",
       "9                  0.7390  ...                 0.8026                 0.7515   \n",
       "10                 0.3024  ...                 0.2004                 0.2715   \n",
       "11                 0.8853  ...                 0.9803                 0.9080   \n",
       "12                 0.6049  ...                 0.7826                 0.6405   \n",
       "13                 0.2678  ...                 0.4973                 0.2197   \n",
       "14                 0.9373  ...                 1.0000                 0.9785   \n",
       "15                 0.1150  ...                 0.0296                 0.1500   \n",
       "16                 0.2891  ...                 0.5110                 0.4035   \n",
       "17                 0.1179  ...                 0.0099                 0.2025   \n",
       "18                 0.6162  ...                 0.5959                 0.5326   \n",
       "19                 0.4842  ...                 0.5454                 0.5401   \n",
       "20                 0.6796  ...                 0.6897                 0.5682   \n",
       "21                 0.8729  ...                 0.9211                 0.8947   \n",
       "\n",
       "   gnomad40_genome_AF_asj gnomad40_genome_AF_eas gnomad40_genome_AF_fin  \\\n",
       "0                  0.0009                 0.0825                 0.0416   \n",
       "1                  0.2778                 0.6203                 0.2227   \n",
       "2                  0.0902                 0.4773                 0.0914   \n",
       "3                  0.0815                 0.1542                 0.0953   \n",
       "4                  0.2412                 0.7667                 0.2028   \n",
       "5                  0.6040                 0.7428                 0.6284   \n",
       "6                  0.4821                 0.5804                 0.4855   \n",
       "7                  0.4763                 0.5740                 0.4785   \n",
       "8                  0.2896                 0.0632                 0.1215   \n",
       "9                  0.5765                 0.9641                 0.7572   \n",
       "10                 0.2656                 0.2284                 0.2924   \n",
       "11                 0.9052                 0.9992                 0.9072   \n",
       "12                 0.6409                 0.6550                 0.6208   \n",
       "13                 0.3844                 0.1176                 0.3131   \n",
       "14                 0.9997                 1.0000                 1.0000   \n",
       "15                 0.0300                 0.1072                 0.0296   \n",
       "16                 0.3041                 0.4909                 0.3275   \n",
       "17                 0.0582                 0.2114                 0.0157   \n",
       "18                 0.5555                 0.7077                 0.5420   \n",
       "19                 0.5972                 0.6021                 0.4084   \n",
       "20                 0.7046                 0.5765                 0.7013   \n",
       "21                 0.8616                 0.9013                 0.8883   \n",
       "\n",
       "   gnomad40_genome_AF_mid gnomad40_genome_AF_nfe gnomad40_genome_AF_remaining  \\\n",
       "0                  0.0204                 0.0100                       0.0592   \n",
       "1                  0.2789                 0.2198                       0.3581   \n",
       "2                  0.0952                 0.0532                       0.0942   \n",
       "3                  0.1748                 0.0724                       0.1162   \n",
       "4                  0.3462                 0.2119                       0.3483   \n",
       "5                  0.5951                 0.5841                       0.6357   \n",
       "6                  0.5602                 0.5225                       0.5178   \n",
       "7                  0.5639                 0.5173                       0.5159   \n",
       "8                  0.2478                 0.1878                       0.2453   \n",
       "9                  0.7279                 0.6921                       0.7277   \n",
       "10                 0.2245                 0.2733                       0.2950   \n",
       "11                 0.9728                 0.8647                       0.9001   \n",
       "12                 0.6103                 0.6411                       0.6187   \n",
       "13                 0.2375                 0.3585                       0.2695   \n",
       "14                 0.9726                 0.9991                       0.9592   \n",
       "15                 0.0714                 0.0253                       0.0965   \n",
       "16                 0.3095                 0.3553                       0.2964   \n",
       "17                 0.0782                 0.0280                       0.1117   \n",
       "18                 0.6016                 0.5871                       0.6202   \n",
       "19                 0.5612                 0.4799                       0.5266   \n",
       "20                 0.7075                 0.6644                       0.6430   \n",
       "21                 0.8571                 0.8847                       0.8646   \n",
       "\n",
       "   gnomad40_genome_AF_sas                    anno  \n",
       "0                  0.0170        Name=11_Weak_Txn  \n",
       "1                  0.3486  Name=13_Heterochrom/lo  \n",
       "2                  0.2200  Name=5_Strong_Enhancer  \n",
       "3                  0.1741  Name=13_Heterochrom/lo  \n",
       "4                  0.4103  Name=13_Heterochrom/lo  \n",
       "5                  0.6683  Name=13_Heterochrom/lo  \n",
       "6                  0.4364  Name=13_Heterochrom/lo  \n",
       "7                  0.4282  Name=13_Heterochrom/lo  \n",
       "8                  0.2055  Name=13_Heterochrom/lo  \n",
       "9                  0.8335  Name=13_Heterochrom/lo  \n",
       "10                 0.3244  Name=13_Heterochrom/lo  \n",
       "11                 0.9679  Name=13_Heterochrom/lo  \n",
       "12                 0.6790  Name=13_Heterochrom/lo  \n",
       "13                 0.2663  Name=13_Heterochrom/lo  \n",
       "14                 0.9992  Name=13_Heterochrom/lo  \n",
       "15                 0.2432  Name=13_Heterochrom/lo  \n",
       "16                 0.3394  Name=13_Heterochrom/lo  \n",
       "17                 0.0729  Name=13_Heterochrom/lo  \n",
       "18                 0.6602  Name=13_Heterochrom/lo  \n",
       "19                 0.5027  Name=13_Heterochrom/lo  \n",
       "20                 0.6366  Name=13_Heterochrom/lo  \n",
       "21                 0.7897  Name=13_Heterochrom/lo  \n",
       "\n",
       "[22 rows x 28 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr22</td>\n",
       "      <td>10666528</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.015013</td>\n",
       "      <td>95.525024</td>\n",
       "      <td>0.055019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr22</td>\n",
       "      <td>10669491</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.014766</td>\n",
       "      <td>68.328550</td>\n",
       "      <td>0.035572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr22</td>\n",
       "      <td>10669673</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.014005</td>\n",
       "      <td>211.582400</td>\n",
       "      <td>0.103881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr22</td>\n",
       "      <td>10672851</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.022202</td>\n",
       "      <td>121.981520</td>\n",
       "      <td>0.062912</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr22</td>\n",
       "      <td>10676441</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.015976</td>\n",
       "      <td>62.980877</td>\n",
       "      <td>0.028924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1089</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50373191</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.015762</td>\n",
       "      <td>93.490160</td>\n",
       "      <td>0.044370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1090</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50542488</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.019690</td>\n",
       "      <td>123.325226</td>\n",
       "      <td>0.060192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1091</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50652342</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.018280</td>\n",
       "      <td>140.713350</td>\n",
       "      <td>0.075278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1092</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50753050</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.017330</td>\n",
       "      <td>69.780640</td>\n",
       "      <td>0.033285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50760504</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.014395</td>\n",
       "      <td>62.968180</td>\n",
       "      <td>0.030079</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1094 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr       pos ref alt  log_counts_diff  log_probs_diff_abs_sum  \\\n",
       "0     chr22  10666528   T   C         0.015013               95.525024   \n",
       "1     chr22  10669491   T   C         0.014766               68.328550   \n",
       "2     chr22  10669673   T   C         0.014005              211.582400   \n",
       "3     chr22  10672851   T   C         0.022202              121.981520   \n",
       "4     chr22  10676441   A   C         0.015976               62.980877   \n",
       "...     ...       ...  ..  ..              ...                     ...   \n",
       "1089  chr22  50373191   T   C         0.015762               93.490160   \n",
       "1090  chr22  50542488   T   C         0.019690              123.325226   \n",
       "1091  chr22  50652342   A   G         0.018280              140.713350   \n",
       "1092  chr22  50753050   T   C         0.017330               69.780640   \n",
       "1093  chr22  50760504   T   C         0.014395               62.968180   \n",
       "\n",
       "      probs_jsd_diff  \n",
       "0           0.055019  \n",
       "1           0.035572  \n",
       "2           0.103881  \n",
       "3           0.062912  \n",
       "4           0.028924  \n",
       "...              ...  \n",
       "1089        0.044370  \n",
       "1090        0.060192  \n",
       "1091        0.075278  \n",
       "1092        0.033285  \n",
       "1093        0.030079  \n",
       "\n",
       "[1094 rows x 7 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_co"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>74092</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.108516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>74138</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.001281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>74837</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000531</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>75025</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>75954</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.004226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15546</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248921925</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15547</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248922537</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15548</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248923056</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15549</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248923390</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15550</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248923401</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000174</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15551 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr        pos ref alt  allele_freq\n",
       "0      chr1      74092   G   A     0.108516\n",
       "1      chr1      74138   T   A     0.001281\n",
       "2      chr1      74837   G   A     0.000531\n",
       "3      chr1      75025   G   A     0.000373\n",
       "4      chr1      75954   C   T     0.004226\n",
       "...     ...        ...  ..  ..          ...\n",
       "15546  chr1  248921925   G   A     0.000261\n",
       "15547  chr1  248922537   T   C     0.000261\n",
       "15548  chr1  248923056   G   T     0.000435\n",
       "15549  chr1  248923390   T   A     0.000174\n",
       "15550  chr1  248923401   C   A     0.000174\n",
       "\n",
       "[15551 rows x 5 columns]"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_af"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>77502</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>86065</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.069599</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>883806</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>894725</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>988951</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6825</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248877982</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6826</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248879155</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6827</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248885752</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6828</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248888164</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.002261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6829</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248893146</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6830 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr        pos ref alt  allele_freq\n",
       "0     chr1      77502   A   G     0.000761\n",
       "1     chr1      86065   G   C     0.069599\n",
       "2     chr1     883806   T   C     0.001913\n",
       "3     chr1     894725   T   C     0.000261\n",
       "4     chr1     988951   T   C     0.000174\n",
       "...    ...        ...  ..  ..          ...\n",
       "6825  chr1  248877982   A   C     0.002087\n",
       "6826  chr1  248879155   T   C     0.002696\n",
       "6827  chr1  248885752   T   C     0.000696\n",
       "6828  chr1  248888164   A   G     0.002261\n",
       "6829  chr1  248893146   A   C     0.000261\n",
       "\n",
       "[6830 rows x 5 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    " data_af"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n"
     ]
    }
   ],
   "source": [
    "i = 1\n",
    "while i <= 22:\n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    print(i)\n",
    "    i += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
