{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import re\n",
    "import gc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2662"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "directory=\"/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model2/anno_af_chrom3/\"\n",
    "# directory2='/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model1/anno_af_chrom/'\n",
    "directory3='/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model2/final_data/'\n",
    "files=os.listdir('model2/anno_af_chrom3')\n",
    "headers=pd.read_csv(\"sampleids_new_sorted.txt\",' ',names=['id','tissue'])\n",
    "for j in range(1,23):\n",
    "    files2=[]\n",
    "    print(j)\n",
    "    for x in files:\n",
    "        if x.startswith(f'gene_chr{j}_'):\n",
    "            files2.append(x)\n",
    "            #allele_freq_chr19_ncr.tsv_exon_chr_19.csv\n",
    "    file=f\"allele_freq_chr{j}_gene_ncr.tsv_chr_{j}.csv\"\n",
    "    snp_data=pd.read_csv(\"/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model2/anno_af_chrom_new/\"+file)\n",
    "    \n",
    "    for i in range(1,121):\n",
    "        #print(i)\n",
    "        \n",
    "        data=pd.read_csv(\"model2/anno_af_chrom3/\"+files2[i],'\\t',names=['chr','pos','end','chr_anno','start_anno','end_anno','anno'])\n",
    "        \n",
    "        \n",
    "        nam=re.search(\"_E\\d*_\",files2[i])[0]\n",
    "        nam=nam.strip(\"_\")\n",
    "        tiss_nam=headers[headers.id==nam]['tissue'].values[0]\n",
    "        #print(tiss_nam)\n",
    "\n",
    "        snp_data=pd.merge(snp_data, data[[\"pos\",\"anno\"]], on=['pos'], how='right')\n",
    "        snp_data=snp_data.rename(columns={\"anno\":tiss_nam})\n",
    "        snp_data_new=snp_data.drop_duplicates(subset=[\"pos\"], keep=False)\n",
    "        snp_data = snp_data_new.copy()\n",
    "    snp_data.to_csv(os.path.join(directory3, 'gene_chr'+str(j)+'.csv'),index=None)\n",
    "        #print(l)\n",
    "        #print(l,files2[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>end</th>\n",
       "      <th>chr_anno</th>\n",
       "      <th>start_anno</th>\n",
       "      <th>end_anno</th>\n",
       "      <th>anno</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3</td>\n",
       "      <td>32434</td>\n",
       "      <td>32434</td>\n",
       "      <td>chr3</td>\n",
       "      <td>0</td>\n",
       "      <td>85200</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3</td>\n",
       "      <td>47480</td>\n",
       "      <td>47480</td>\n",
       "      <td>chr3</td>\n",
       "      <td>0</td>\n",
       "      <td>85200</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3</td>\n",
       "      <td>48195</td>\n",
       "      <td>48195</td>\n",
       "      <td>chr3</td>\n",
       "      <td>0</td>\n",
       "      <td>85200</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3</td>\n",
       "      <td>48202</td>\n",
       "      <td>48202</td>\n",
       "      <td>chr3</td>\n",
       "      <td>0</td>\n",
       "      <td>85200</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3</td>\n",
       "      <td>60400</td>\n",
       "      <td>60400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>0</td>\n",
       "      <td>85200</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7031</th>\n",
       "      <td>chr3</td>\n",
       "      <td>197480734</td>\n",
       "      <td>197480734</td>\n",
       "      <td>chr3</td>\n",
       "      <td>197480000</td>\n",
       "      <td>197488000</td>\n",
       "      <td>5_TxWk</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7032</th>\n",
       "      <td>chr3</td>\n",
       "      <td>197507547</td>\n",
       "      <td>197507547</td>\n",
       "      <td>chr3</td>\n",
       "      <td>197507400</td>\n",
       "      <td>197509000</td>\n",
       "      <td>5_TxWk</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7033</th>\n",
       "      <td>chr3</td>\n",
       "      <td>197576155</td>\n",
       "      <td>197576155</td>\n",
       "      <td>chr3</td>\n",
       "      <td>197576000</td>\n",
       "      <td>197579400</td>\n",
       "      <td>5_TxWk</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7034</th>\n",
       "      <td>chr3</td>\n",
       "      <td>197636492</td>\n",
       "      <td>197636492</td>\n",
       "      <td>chr3</td>\n",
       "      <td>197633600</td>\n",
       "      <td>197639600</td>\n",
       "      <td>14_ReprPCWk</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7035</th>\n",
       "      <td>chr3</td>\n",
       "      <td>197656793</td>\n",
       "      <td>197656793</td>\n",
       "      <td>chr3</td>\n",
       "      <td>197640000</td>\n",
       "      <td>197675400</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7036 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr        pos        end chr_anno  start_anno   end_anno         anno\n",
       "0     chr3      32434      32434     chr3           0      85200     15_Quies\n",
       "1     chr3      47480      47480     chr3           0      85200     15_Quies\n",
       "2     chr3      48195      48195     chr3           0      85200     15_Quies\n",
       "3     chr3      48202      48202     chr3           0      85200     15_Quies\n",
       "4     chr3      60400      60400     chr3           0      85200     15_Quies\n",
       "...    ...        ...        ...      ...         ...        ...          ...\n",
       "7031  chr3  197480734  197480734     chr3   197480000  197488000       5_TxWk\n",
       "7032  chr3  197507547  197507547     chr3   197507400  197509000       5_TxWk\n",
       "7033  chr3  197576155  197576155     chr3   197576000  197579400       5_TxWk\n",
       "7034  chr3  197636492  197636492     chr3   197633600  197639600  14_ReprPCWk\n",
       "7035  chr3  197656793  197656793     chr3   197640000  197675400     15_Quies\n",
       "\n",
       "[7036 rows x 7 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "headers=pd.read_csv(\"sampleids_new_sorted.txt\",' ',names=['id','tissue'])\n",
    "\n",
    "    files2=[]\n",
    "    for x in files:\n",
    "        if x.startswith(f'gene_chr{j}_'):\n",
    "            files2.append(x)\n",
    "            #allele_freq_chr19_ncr.tsv_exon_chr_19.csv\n",
    "    file=f\"allele_freq_chr{j}_gene_ncr.tsv_chr_{j}.csv\"\n",
    "    snp_data=pd.read_csv(\"/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model1/anno_af_chrom_new/\"+file)\n",
    "    \n",
    "    for i in range(121):\n",
    "        print(i)\n",
    "        data=pd.read_csv(\"model1/anno_af_chrom3/\"+files2[i],'\\t',names=['chr','pos','end','chr_anno','start_anno','end_anno','anno'])\n",
    "        nam=re.search(\"_E\\d*_\",files2[i])[0]\n",
    "        nam=nam.strip(\"_\")\n",
    "        tiss_nam=headers[headers.id==nam]['tissue'].values[0]\n",
    "        #print(tiss_nam)\n",
    "\n",
    "        snp_data.set_index('pos', inplace=True)\n",
    "        data.set_index('pos', inplace=True)\n",
    "        snp_data = pd.merge(snp_data, data[[\"anno\"]], left_index=True, right_index=True, how='left')\n",
    "        snp_data.reset_index(inplace=True)\n",
    "                \n",
    "        #snp_data=pd.merge(snp_data, data[[\"pos\",\"anno\"]], on=['pos'], how='left')\n",
    "        snp_data=snp_data.rename(columns={\"anno\":tiss_nam})\n",
    "    snp_data=snp_data.drop_duplicates(subset=[\"pos\"], keep=False)\n",
    "    snp_data.to_csv(os.path.join(directory3, 'gene_chr'+str(j)+'.csv'),index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import re\n",
    "import gc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "files=os.listdir('model1/anno_af_chrom3')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "headers=pd.read_csv(\"sampleids_new_sorted.txt\",' ',names=['id','tissue'])\n",
    "\n",
    "j=3\n",
    "files2=[]\n",
    "for x in files:\n",
    "    if x.startswith(f'gene_chr{j}_'):\n",
    "        files2.append(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2663"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "file=f\"allele_freq_chr{j}_gene_ncr.tsv_chr_{j}.csv\"\n",
    "snp_data=pd.read_csv(\"/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model1/anno_af_chrom_new/\"+file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_data_head=snp_data.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7037, 27)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#snp_data.to_csv(os.path.join(directory3, 'gene_chr'+str(j)+'check_old.csv'),index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "i=90\n",
    "data=pd.read_csv(\"model1/anno_af_chrom3/\"+files2[i],'\\t',names=['chr','pos','end','chr_anno','start_anno','end_anno','anno'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "nam=re.search(\"_E\\d*_\",files2[i])[0]\n",
    "nam=nam.strip(\"_\")\n",
    "tiss_nam=headers[headers.id==nam]['tissue'].values[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NH-A_Astrocytes'"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tiss_nam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_data.set_index('pos', inplace=True)\n",
    "data.set_index('pos', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr_x</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>gnomad40_genome_AF_XX</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad40_genome_AF_afr</th>\n",
       "      <th>gnomad40_genome_AF_ami</th>\n",
       "      <th>gnomad40_genome_AF_amr</th>\n",
       "      <th>gnomad40_genome_AF_asj</th>\n",
       "      <th>gnomad40_genome_AF_eas</th>\n",
       "      <th>gnomad40_genome_AF_fin</th>\n",
       "      <th>gnomad40_genome_AF_mid</th>\n",
       "      <th>gnomad40_genome_AF_nfe</th>\n",
       "      <th>gnomad40_genome_AF_remaining</th>\n",
       "      <th>gnomad40_genome_AF_sas</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pos</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>32434</th>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.018746</td>\n",
       "      <td>165.452320</td>\n",
       "      <td>0.097936</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47480</th>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015024</td>\n",
       "      <td>126.894470</td>\n",
       "      <td>0.070111</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>8.993e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48195</th>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.019221</td>\n",
       "      <td>0.015570</td>\n",
       "      <td>77.986465</td>\n",
       "      <td>0.043584</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>...</td>\n",
       "      <td>2.407e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.411e-05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.0182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48202</th>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.016036</td>\n",
       "      <td>115.160090</td>\n",
       "      <td>0.067440</td>\n",
       "      <td>3.284e-05</td>\n",
       "      <td>3.281e-05</td>\n",
       "      <td>3.855e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60400</th>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>0.015248</td>\n",
       "      <td>79.551820</td>\n",
       "      <td>0.037764</td>\n",
       "      <td>1.314e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197576155</th>\n",
       "      <td>chr3</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015796</td>\n",
       "      <td>94.429090</td>\n",
       "      <td>0.051905</td>\n",
       "      <td>2.74e-05</td>\n",
       "      <td>2.684e-05</td>\n",
       "      <td>5.326e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>5.208e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.484e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197636492</th>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015411</td>\n",
       "      <td>62.338760</td>\n",
       "      <td>0.026065</td>\n",
       "      <td>8.088e-05</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>3.943e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>5.003e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>6.805e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197656793</th>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.055652</td>\n",
       "      <td>0.017351</td>\n",
       "      <td>84.323980</td>\n",
       "      <td>0.037448</td>\n",
       "      <td>0.1119</td>\n",
       "      <td>0.1683</td>\n",
       "      <td>0.1197</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3134</td>\n",
       "      <td>0.1662</td>\n",
       "      <td>0.0692</td>\n",
       "      <td>0.0709</td>\n",
       "      <td>0.0022</td>\n",
       "      <td>0.0349</td>\n",
       "      <td>0.1206</td>\n",
       "      <td>0.0602</td>\n",
       "      <td>0.0888</td>\n",
       "      <td>0.0348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198116021</th>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.241652</td>\n",
       "      <td>0.018788</td>\n",
       "      <td>87.795296</td>\n",
       "      <td>0.039123</td>\n",
       "      <td>0.2093</td>\n",
       "      <td>0.2097</td>\n",
       "      <td>0.2069</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.0647</td>\n",
       "      <td>0.1589</td>\n",
       "      <td>0.1773</td>\n",
       "      <td>0.0702</td>\n",
       "      <td>0.2510</td>\n",
       "      <td>0.1939</td>\n",
       "      <td>0.1599</td>\n",
       "      <td>0.1744</td>\n",
       "      <td>0.2595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198117279</th>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.016466</td>\n",
       "      <td>106.986080</td>\n",
       "      <td>0.056975</td>\n",
       "      <td>0.0301</td>\n",
       "      <td>0.1578</td>\n",
       "      <td>0.0294</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0456</td>\n",
       "      <td>0.0062</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.0201</td>\n",
       "      <td>0.0087</td>\n",
       "      <td>0.0598</td>\n",
       "      <td>0.0287</td>\n",
       "      <td>0.0198</td>\n",
       "      <td>0.0278</td>\n",
       "      <td>0.0359</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7037 rows Ã— 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          chr_x ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "pos                                                         \n",
       "32434      chr3     A     C     0.000261         0.018746   \n",
       "47480      chr3     T     C     0.000174         0.015024   \n",
       "48195      chr3     T     C     0.019221         0.015570   \n",
       "48202      chr3     G     C     0.001739         0.016036   \n",
       "60400      chr3     G     C     0.001044         0.015248   \n",
       "...         ...   ...   ...          ...              ...   \n",
       "197576155  chr3     C     G     0.000174         0.015796   \n",
       "197636492  chr3     T     C     0.000174         0.015411   \n",
       "197656793  chr3     T     C     0.055652         0.017351   \n",
       "198116021  chr3     T     C     0.241652         0.018788   \n",
       "198117279  chr3     T     C     0.000177         0.016466   \n",
       "\n",
       "           log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "pos                                                                    \n",
       "32434                  165.452320        0.097936                  .   \n",
       "47480                  126.894470        0.070111             0.0001   \n",
       "48195                   77.986465        0.043584             0.0006   \n",
       "48202                  115.160090        0.067440          3.284e-05   \n",
       "60400                   79.551820        0.037764          1.314e-05   \n",
       "...                           ...             ...                ...   \n",
       "197576155               94.429090        0.051905           2.74e-05   \n",
       "197636492               62.338760        0.026065          8.088e-05   \n",
       "197656793               84.323980        0.037448             0.1119   \n",
       "198116021               87.795296        0.039123             0.2093   \n",
       "198117279              106.986080        0.056975             0.0301   \n",
       "\n",
       "          gnomad40_genome_AF_raw gnomad40_genome_AF_XX  ...  \\\n",
       "pos                                                     ...   \n",
       "32434                          .                     .  ...   \n",
       "47480                     0.0001             8.993e-05  ...   \n",
       "48195                     0.0006                0.0003  ...   \n",
       "48202                  3.281e-05             3.855e-05  ...   \n",
       "60400                  1.312e-05                     0  ...   \n",
       "...                          ...                   ...  ...   \n",
       "197576155              2.684e-05             5.326e-05  ...   \n",
       "197636492                 0.0011             3.943e-05  ...   \n",
       "197656793                 0.1683                0.1197  ...   \n",
       "198116021                 0.2097                0.2069  ...   \n",
       "198117279                 0.1578                0.0294  ...   \n",
       "\n",
       "          gnomad40_genome_AF_afr gnomad40_genome_AF_ami  \\\n",
       "pos                                                       \n",
       "32434                          .                      .   \n",
       "47480                     0.0004                      0   \n",
       "48195                  2.407e-05                      0   \n",
       "48202                          0                      0   \n",
       "60400                          0                      0   \n",
       "...                          ...                    ...   \n",
       "197576155              5.208e-05                      0   \n",
       "197636492              5.003e-05                      0   \n",
       "197656793                 0.3134                 0.1662   \n",
       "198116021                 0.3173                 0.0647   \n",
       "198117279                 0.0456                 0.0062   \n",
       "\n",
       "          gnomad40_genome_AF_amr gnomad40_genome_AF_asj  \\\n",
       "pos                                                       \n",
       "32434                          .                      .   \n",
       "47480                     0.0001                      0   \n",
       "48195                     0.0001                      0   \n",
       "48202                          0                      0   \n",
       "60400                          0                      0   \n",
       "...                          ...                    ...   \n",
       "197576155                      0                      0   \n",
       "197636492              6.805e-05                      0   \n",
       "197656793                 0.0692                 0.0709   \n",
       "198116021                 0.1589                 0.1773   \n",
       "198117279                 0.0308                 0.0201   \n",
       "\n",
       "          gnomad40_genome_AF_eas gnomad40_genome_AF_fin  \\\n",
       "pos                                                       \n",
       "32434                          .                      .   \n",
       "47480                          0                      0   \n",
       "48195                          0                      0   \n",
       "48202                          0                      0   \n",
       "60400                     0.0002                      0   \n",
       "...                          ...                    ...   \n",
       "197576155                      0                      0   \n",
       "197636492                 0.0002                      0   \n",
       "197656793                 0.0022                 0.0349   \n",
       "198116021                 0.0702                 0.2510   \n",
       "198117279                 0.0087                 0.0598   \n",
       "\n",
       "          gnomad40_genome_AF_mid gnomad40_genome_AF_nfe  \\\n",
       "pos                                                       \n",
       "32434                          .                      .   \n",
       "47480                          0                      0   \n",
       "48195                          0              4.411e-05   \n",
       "48202                          0               1.47e-05   \n",
       "60400                          0                      0   \n",
       "...                          ...                    ...   \n",
       "197576155                      0              1.484e-05   \n",
       "197636492                      0                 0.0001   \n",
       "197656793                 0.1206                 0.0602   \n",
       "198116021                 0.1939                 0.1599   \n",
       "198117279                 0.0287                 0.0198   \n",
       "\n",
       "          gnomad40_genome_AF_remaining gnomad40_genome_AF_sas  \n",
       "pos                                                            \n",
       "32434                                .                      .  \n",
       "47480                                0                      0  \n",
       "48195                           0.0009                 0.0182  \n",
       "48202                                0                 0.0008  \n",
       "60400                                0                 0.0002  \n",
       "...                                ...                    ...  \n",
       "197576155                            0                 0.0002  \n",
       "197636492                            0                      0  \n",
       "197656793                       0.0888                 0.0348  \n",
       "198116021                       0.1744                 0.2595  \n",
       "198117279                       0.0278                 0.0359  \n",
       "\n",
       "[7037 rows x 26 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_data_new = pd.merge(snp_data, data[[\"anno\"]], left_index=True, right_index=True, how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_data_new.reset_index(inplace=True)\n",
    "#data.reset_index(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_data.reset_index(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos</th>\n",
       "      <th>chr_x</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad40_genome_AF_afr</th>\n",
       "      <th>gnomad40_genome_AF_ami</th>\n",
       "      <th>gnomad40_genome_AF_amr</th>\n",
       "      <th>gnomad40_genome_AF_asj</th>\n",
       "      <th>gnomad40_genome_AF_eas</th>\n",
       "      <th>gnomad40_genome_AF_fin</th>\n",
       "      <th>gnomad40_genome_AF_mid</th>\n",
       "      <th>gnomad40_genome_AF_nfe</th>\n",
       "      <th>gnomad40_genome_AF_remaining</th>\n",
       "      <th>gnomad40_genome_AF_sas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>32434</td>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.018746</td>\n",
       "      <td>165.452320</td>\n",
       "      <td>0.097936</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>47480</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015024</td>\n",
       "      <td>126.894470</td>\n",
       "      <td>0.070111</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>48195</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.019221</td>\n",
       "      <td>0.015570</td>\n",
       "      <td>77.986465</td>\n",
       "      <td>0.043584</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>...</td>\n",
       "      <td>2.407e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.411e-05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.0182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>48202</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.016036</td>\n",
       "      <td>115.160090</td>\n",
       "      <td>0.067440</td>\n",
       "      <td>3.284e-05</td>\n",
       "      <td>3.281e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>60400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>0.015248</td>\n",
       "      <td>79.551820</td>\n",
       "      <td>0.037764</td>\n",
       "      <td>1.314e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7032</th>\n",
       "      <td>197576155</td>\n",
       "      <td>chr3</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015796</td>\n",
       "      <td>94.429090</td>\n",
       "      <td>0.051905</td>\n",
       "      <td>2.74e-05</td>\n",
       "      <td>2.684e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>5.208e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.484e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7033</th>\n",
       "      <td>197636492</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015411</td>\n",
       "      <td>62.338760</td>\n",
       "      <td>0.026065</td>\n",
       "      <td>8.088e-05</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>...</td>\n",
       "      <td>5.003e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>6.805e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7034</th>\n",
       "      <td>197656793</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.055652</td>\n",
       "      <td>0.017351</td>\n",
       "      <td>84.323980</td>\n",
       "      <td>0.037448</td>\n",
       "      <td>0.1119</td>\n",
       "      <td>0.1683</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3134</td>\n",
       "      <td>0.1662</td>\n",
       "      <td>0.0692</td>\n",
       "      <td>0.0709</td>\n",
       "      <td>0.0022</td>\n",
       "      <td>0.0349</td>\n",
       "      <td>0.1206</td>\n",
       "      <td>0.0602</td>\n",
       "      <td>0.0888</td>\n",
       "      <td>0.0348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7035</th>\n",
       "      <td>198116021</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.241652</td>\n",
       "      <td>0.018788</td>\n",
       "      <td>87.795296</td>\n",
       "      <td>0.039123</td>\n",
       "      <td>0.2093</td>\n",
       "      <td>0.2097</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.0647</td>\n",
       "      <td>0.1589</td>\n",
       "      <td>0.1773</td>\n",
       "      <td>0.0702</td>\n",
       "      <td>0.2510</td>\n",
       "      <td>0.1939</td>\n",
       "      <td>0.1599</td>\n",
       "      <td>0.1744</td>\n",
       "      <td>0.2595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7036</th>\n",
       "      <td>198117279</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.016466</td>\n",
       "      <td>106.986080</td>\n",
       "      <td>0.056975</td>\n",
       "      <td>0.0301</td>\n",
       "      <td>0.1578</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0456</td>\n",
       "      <td>0.0062</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.0201</td>\n",
       "      <td>0.0087</td>\n",
       "      <td>0.0598</td>\n",
       "      <td>0.0287</td>\n",
       "      <td>0.0198</td>\n",
       "      <td>0.0278</td>\n",
       "      <td>0.0359</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7037 rows Ã— 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pos chr_x ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0         32434  chr3     A     C     0.000261         0.018746   \n",
       "1         47480  chr3     T     C     0.000174         0.015024   \n",
       "2         48195  chr3     T     C     0.019221         0.015570   \n",
       "3         48202  chr3     G     C     0.001739         0.016036   \n",
       "4         60400  chr3     G     C     0.001044         0.015248   \n",
       "...         ...   ...   ...   ...          ...              ...   \n",
       "7032  197576155  chr3     C     G     0.000174         0.015796   \n",
       "7033  197636492  chr3     T     C     0.000174         0.015411   \n",
       "7034  197656793  chr3     T     C     0.055652         0.017351   \n",
       "7035  198116021  chr3     T     C     0.241652         0.018788   \n",
       "7036  198117279  chr3     T     C     0.000177         0.016466   \n",
       "\n",
       "      log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                 165.452320        0.097936                  .   \n",
       "1                 126.894470        0.070111             0.0001   \n",
       "2                  77.986465        0.043584             0.0006   \n",
       "3                 115.160090        0.067440          3.284e-05   \n",
       "4                  79.551820        0.037764          1.314e-05   \n",
       "...                      ...             ...                ...   \n",
       "7032               94.429090        0.051905           2.74e-05   \n",
       "7033               62.338760        0.026065          8.088e-05   \n",
       "7034               84.323980        0.037448             0.1119   \n",
       "7035               87.795296        0.039123             0.2093   \n",
       "7036              106.986080        0.056975             0.0301   \n",
       "\n",
       "     gnomad40_genome_AF_raw  ... gnomad40_genome_AF_afr  \\\n",
       "0                         .  ...                      .   \n",
       "1                    0.0001  ...                 0.0004   \n",
       "2                    0.0006  ...              2.407e-05   \n",
       "3                 3.281e-05  ...                      0   \n",
       "4                 1.312e-05  ...                      0   \n",
       "...                     ...  ...                    ...   \n",
       "7032              2.684e-05  ...              5.208e-05   \n",
       "7033                 0.0011  ...              5.003e-05   \n",
       "7034                 0.1683  ...                 0.3134   \n",
       "7035                 0.2097  ...                 0.3173   \n",
       "7036                 0.1578  ...                 0.0456   \n",
       "\n",
       "     gnomad40_genome_AF_ami gnomad40_genome_AF_amr gnomad40_genome_AF_asj  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                 0.0001                      0   \n",
       "2                         0                 0.0001                      0   \n",
       "3                         0                      0                      0   \n",
       "4                         0                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "7032                      0                      0                      0   \n",
       "7033                      0              6.805e-05                      0   \n",
       "7034                 0.1662                 0.0692                 0.0709   \n",
       "7035                 0.0647                 0.1589                 0.1773   \n",
       "7036                 0.0062                 0.0308                 0.0201   \n",
       "\n",
       "     gnomad40_genome_AF_eas gnomad40_genome_AF_fin gnomad40_genome_AF_mid  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                      0                      0   \n",
       "2                         0                      0                      0   \n",
       "3                         0                      0                      0   \n",
       "4                    0.0002                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "7032                      0                      0                      0   \n",
       "7033                 0.0002                      0                      0   \n",
       "7034                 0.0022                 0.0349                 0.1206   \n",
       "7035                 0.0702                 0.2510                 0.1939   \n",
       "7036                 0.0087                 0.0598                 0.0287   \n",
       "\n",
       "     gnomad40_genome_AF_nfe gnomad40_genome_AF_remaining  \\\n",
       "0                         .                            .   \n",
       "1                         0                            0   \n",
       "2                 4.411e-05                       0.0009   \n",
       "3                  1.47e-05                            0   \n",
       "4                         0                            0   \n",
       "...                     ...                          ...   \n",
       "7032              1.484e-05                            0   \n",
       "7033                 0.0001                            0   \n",
       "7034                 0.0602                       0.0888   \n",
       "7035                 0.1599                       0.1744   \n",
       "7036                 0.0198                       0.0278   \n",
       "\n",
       "     gnomad40_genome_AF_sas  \n",
       "0                         .  \n",
       "1                         0  \n",
       "2                    0.0182  \n",
       "3                    0.0008  \n",
       "4                    0.0002  \n",
       "...                     ...  \n",
       "7032                 0.0002  \n",
       "7033                      0  \n",
       "7034                 0.0348  \n",
       "7035                 0.2595  \n",
       "7036                 0.0359  \n",
       "\n",
       "[7037 rows x 27 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "#snp_data=pd.merge(snp_data, data[[\"pos\",\"anno\"]], on=['pos'], how='left')\n",
    "snp_data_new=snp_data.rename(columns={\"anno\":tiss_nam})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos</th>\n",
       "      <th>chr_x</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad40_genome_AF_ami</th>\n",
       "      <th>gnomad40_genome_AF_amr</th>\n",
       "      <th>gnomad40_genome_AF_asj</th>\n",
       "      <th>gnomad40_genome_AF_eas</th>\n",
       "      <th>gnomad40_genome_AF_fin</th>\n",
       "      <th>gnomad40_genome_AF_mid</th>\n",
       "      <th>gnomad40_genome_AF_nfe</th>\n",
       "      <th>gnomad40_genome_AF_remaining</th>\n",
       "      <th>gnomad40_genome_AF_sas</th>\n",
       "      <th>anno</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>32434</td>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.018746</td>\n",
       "      <td>165.452320</td>\n",
       "      <td>0.097936</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>47480</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015024</td>\n",
       "      <td>126.894470</td>\n",
       "      <td>0.070111</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>48195</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.019221</td>\n",
       "      <td>0.015570</td>\n",
       "      <td>77.986465</td>\n",
       "      <td>0.043584</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.411e-05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.0182</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>48202</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.016036</td>\n",
       "      <td>115.160090</td>\n",
       "      <td>0.067440</td>\n",
       "      <td>3.284e-05</td>\n",
       "      <td>3.281e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0008</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>60400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>0.015248</td>\n",
       "      <td>79.551820</td>\n",
       "      <td>0.037764</td>\n",
       "      <td>1.314e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7033</th>\n",
       "      <td>197576155</td>\n",
       "      <td>chr3</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015796</td>\n",
       "      <td>94.429090</td>\n",
       "      <td>0.051905</td>\n",
       "      <td>2.74e-05</td>\n",
       "      <td>2.684e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.484e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>4_Tx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7034</th>\n",
       "      <td>197636492</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015411</td>\n",
       "      <td>62.338760</td>\n",
       "      <td>0.026065</td>\n",
       "      <td>8.088e-05</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>6.805e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7035</th>\n",
       "      <td>197656793</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.055652</td>\n",
       "      <td>0.017351</td>\n",
       "      <td>84.323980</td>\n",
       "      <td>0.037448</td>\n",
       "      <td>0.1119</td>\n",
       "      <td>0.1683</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1662</td>\n",
       "      <td>0.0692</td>\n",
       "      <td>0.0709</td>\n",
       "      <td>0.0022</td>\n",
       "      <td>0.0349</td>\n",
       "      <td>0.1206</td>\n",
       "      <td>0.0602</td>\n",
       "      <td>0.0888</td>\n",
       "      <td>0.0348</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7036</th>\n",
       "      <td>198116021</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.241652</td>\n",
       "      <td>0.018788</td>\n",
       "      <td>87.795296</td>\n",
       "      <td>0.039123</td>\n",
       "      <td>0.2093</td>\n",
       "      <td>0.2097</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0647</td>\n",
       "      <td>0.1589</td>\n",
       "      <td>0.1773</td>\n",
       "      <td>0.0702</td>\n",
       "      <td>0.2510</td>\n",
       "      <td>0.1939</td>\n",
       "      <td>0.1599</td>\n",
       "      <td>0.1744</td>\n",
       "      <td>0.2595</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7037</th>\n",
       "      <td>198117279</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.016466</td>\n",
       "      <td>106.986080</td>\n",
       "      <td>0.056975</td>\n",
       "      <td>0.0301</td>\n",
       "      <td>0.1578</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0062</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.0201</td>\n",
       "      <td>0.0087</td>\n",
       "      <td>0.0598</td>\n",
       "      <td>0.0287</td>\n",
       "      <td>0.0198</td>\n",
       "      <td>0.0278</td>\n",
       "      <td>0.0359</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7038 rows Ã— 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pos chr_x ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0         32434  chr3     A     C     0.000261         0.018746   \n",
       "1         47480  chr3     T     C     0.000174         0.015024   \n",
       "2         48195  chr3     T     C     0.019221         0.015570   \n",
       "3         48202  chr3     G     C     0.001739         0.016036   \n",
       "4         60400  chr3     G     C     0.001044         0.015248   \n",
       "...         ...   ...   ...   ...          ...              ...   \n",
       "7033  197576155  chr3     C     G     0.000174         0.015796   \n",
       "7034  197636492  chr3     T     C     0.000174         0.015411   \n",
       "7035  197656793  chr3     T     C     0.055652         0.017351   \n",
       "7036  198116021  chr3     T     C     0.241652         0.018788   \n",
       "7037  198117279  chr3     T     C     0.000177         0.016466   \n",
       "\n",
       "      log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                 165.452320        0.097936                  .   \n",
       "1                 126.894470        0.070111             0.0001   \n",
       "2                  77.986465        0.043584             0.0006   \n",
       "3                 115.160090        0.067440          3.284e-05   \n",
       "4                  79.551820        0.037764          1.314e-05   \n",
       "...                      ...             ...                ...   \n",
       "7033               94.429090        0.051905           2.74e-05   \n",
       "7034               62.338760        0.026065          8.088e-05   \n",
       "7035               84.323980        0.037448             0.1119   \n",
       "7036               87.795296        0.039123             0.2093   \n",
       "7037              106.986080        0.056975             0.0301   \n",
       "\n",
       "     gnomad40_genome_AF_raw  ... gnomad40_genome_AF_ami  \\\n",
       "0                         .  ...                      .   \n",
       "1                    0.0001  ...                      0   \n",
       "2                    0.0006  ...                      0   \n",
       "3                 3.281e-05  ...                      0   \n",
       "4                 1.312e-05  ...                      0   \n",
       "...                     ...  ...                    ...   \n",
       "7033              2.684e-05  ...                      0   \n",
       "7034                 0.0011  ...                      0   \n",
       "7035                 0.1683  ...                 0.1662   \n",
       "7036                 0.2097  ...                 0.0647   \n",
       "7037                 0.1578  ...                 0.0062   \n",
       "\n",
       "     gnomad40_genome_AF_amr gnomad40_genome_AF_asj gnomad40_genome_AF_eas  \\\n",
       "0                         .                      .                      .   \n",
       "1                    0.0001                      0                      0   \n",
       "2                    0.0001                      0                      0   \n",
       "3                         0                      0                      0   \n",
       "4                         0                      0                 0.0002   \n",
       "...                     ...                    ...                    ...   \n",
       "7033                      0                      0                      0   \n",
       "7034              6.805e-05                      0                 0.0002   \n",
       "7035                 0.0692                 0.0709                 0.0022   \n",
       "7036                 0.1589                 0.1773                 0.0702   \n",
       "7037                 0.0308                 0.0201                 0.0087   \n",
       "\n",
       "     gnomad40_genome_AF_fin gnomad40_genome_AF_mid gnomad40_genome_AF_nfe  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                      0                      0   \n",
       "2                         0                      0              4.411e-05   \n",
       "3                         0                      0               1.47e-05   \n",
       "4                         0                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "7033                      0                      0              1.484e-05   \n",
       "7034                      0                      0                 0.0001   \n",
       "7035                 0.0349                 0.1206                 0.0602   \n",
       "7036                 0.2510                 0.1939                 0.1599   \n",
       "7037                 0.0598                 0.0287                 0.0198   \n",
       "\n",
       "     gnomad40_genome_AF_remaining gnomad40_genome_AF_sas      anno  \n",
       "0                               .                      .  15_Quies  \n",
       "1                               0                      0  15_Quies  \n",
       "2                          0.0009                 0.0182  15_Quies  \n",
       "3                               0                 0.0008  15_Quies  \n",
       "4                               0                 0.0002  15_Quies  \n",
       "...                           ...                    ...       ...  \n",
       "7033                            0                 0.0002      4_Tx  \n",
       "7034                            0                      0  15_Quies  \n",
       "7035                       0.0888                 0.0348  15_Quies  \n",
       "7036                       0.1744                 0.2595       NaN  \n",
       "7037                       0.0278                 0.0359       NaN  \n",
       "\n",
       "[7038 rows x 28 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_data_1=snp_data_new.drop_duplicates(subset=[\"pos\"], keep=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos</th>\n",
       "      <th>chr_x</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad40_genome_AF_ami</th>\n",
       "      <th>gnomad40_genome_AF_amr</th>\n",
       "      <th>gnomad40_genome_AF_asj</th>\n",
       "      <th>gnomad40_genome_AF_eas</th>\n",
       "      <th>gnomad40_genome_AF_fin</th>\n",
       "      <th>gnomad40_genome_AF_mid</th>\n",
       "      <th>gnomad40_genome_AF_nfe</th>\n",
       "      <th>gnomad40_genome_AF_remaining</th>\n",
       "      <th>gnomad40_genome_AF_sas</th>\n",
       "      <th>anno</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>32434</td>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.018746</td>\n",
       "      <td>165.452320</td>\n",
       "      <td>0.097936</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>47480</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015024</td>\n",
       "      <td>126.894470</td>\n",
       "      <td>0.070111</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>48195</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.019221</td>\n",
       "      <td>0.015570</td>\n",
       "      <td>77.986465</td>\n",
       "      <td>0.043584</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.411e-05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.0182</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>48202</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.016036</td>\n",
       "      <td>115.160090</td>\n",
       "      <td>0.067440</td>\n",
       "      <td>3.284e-05</td>\n",
       "      <td>3.281e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0008</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>60400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>0.015248</td>\n",
       "      <td>79.551820</td>\n",
       "      <td>0.037764</td>\n",
       "      <td>1.314e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7033</th>\n",
       "      <td>197576155</td>\n",
       "      <td>chr3</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015796</td>\n",
       "      <td>94.429090</td>\n",
       "      <td>0.051905</td>\n",
       "      <td>2.74e-05</td>\n",
       "      <td>2.684e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.484e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>4_Tx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7034</th>\n",
       "      <td>197636492</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015411</td>\n",
       "      <td>62.338760</td>\n",
       "      <td>0.026065</td>\n",
       "      <td>8.088e-05</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>6.805e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7035</th>\n",
       "      <td>197656793</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.055652</td>\n",
       "      <td>0.017351</td>\n",
       "      <td>84.323980</td>\n",
       "      <td>0.037448</td>\n",
       "      <td>0.1119</td>\n",
       "      <td>0.1683</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1662</td>\n",
       "      <td>0.0692</td>\n",
       "      <td>0.0709</td>\n",
       "      <td>0.0022</td>\n",
       "      <td>0.0349</td>\n",
       "      <td>0.1206</td>\n",
       "      <td>0.0602</td>\n",
       "      <td>0.0888</td>\n",
       "      <td>0.0348</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7036</th>\n",
       "      <td>198116021</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.241652</td>\n",
       "      <td>0.018788</td>\n",
       "      <td>87.795296</td>\n",
       "      <td>0.039123</td>\n",
       "      <td>0.2093</td>\n",
       "      <td>0.2097</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0647</td>\n",
       "      <td>0.1589</td>\n",
       "      <td>0.1773</td>\n",
       "      <td>0.0702</td>\n",
       "      <td>0.2510</td>\n",
       "      <td>0.1939</td>\n",
       "      <td>0.1599</td>\n",
       "      <td>0.1744</td>\n",
       "      <td>0.2595</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7037</th>\n",
       "      <td>198117279</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.016466</td>\n",
       "      <td>106.986080</td>\n",
       "      <td>0.056975</td>\n",
       "      <td>0.0301</td>\n",
       "      <td>0.1578</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0062</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.0201</td>\n",
       "      <td>0.0087</td>\n",
       "      <td>0.0598</td>\n",
       "      <td>0.0287</td>\n",
       "      <td>0.0198</td>\n",
       "      <td>0.0278</td>\n",
       "      <td>0.0359</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7036 rows Ã— 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pos chr_x ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0         32434  chr3     A     C     0.000261         0.018746   \n",
       "1         47480  chr3     T     C     0.000174         0.015024   \n",
       "2         48195  chr3     T     C     0.019221         0.015570   \n",
       "3         48202  chr3     G     C     0.001739         0.016036   \n",
       "4         60400  chr3     G     C     0.001044         0.015248   \n",
       "...         ...   ...   ...   ...          ...              ...   \n",
       "7033  197576155  chr3     C     G     0.000174         0.015796   \n",
       "7034  197636492  chr3     T     C     0.000174         0.015411   \n",
       "7035  197656793  chr3     T     C     0.055652         0.017351   \n",
       "7036  198116021  chr3     T     C     0.241652         0.018788   \n",
       "7037  198117279  chr3     T     C     0.000177         0.016466   \n",
       "\n",
       "      log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                 165.452320        0.097936                  .   \n",
       "1                 126.894470        0.070111             0.0001   \n",
       "2                  77.986465        0.043584             0.0006   \n",
       "3                 115.160090        0.067440          3.284e-05   \n",
       "4                  79.551820        0.037764          1.314e-05   \n",
       "...                      ...             ...                ...   \n",
       "7033               94.429090        0.051905           2.74e-05   \n",
       "7034               62.338760        0.026065          8.088e-05   \n",
       "7035               84.323980        0.037448             0.1119   \n",
       "7036               87.795296        0.039123             0.2093   \n",
       "7037              106.986080        0.056975             0.0301   \n",
       "\n",
       "     gnomad40_genome_AF_raw  ... gnomad40_genome_AF_ami  \\\n",
       "0                         .  ...                      .   \n",
       "1                    0.0001  ...                      0   \n",
       "2                    0.0006  ...                      0   \n",
       "3                 3.281e-05  ...                      0   \n",
       "4                 1.312e-05  ...                      0   \n",
       "...                     ...  ...                    ...   \n",
       "7033              2.684e-05  ...                      0   \n",
       "7034                 0.0011  ...                      0   \n",
       "7035                 0.1683  ...                 0.1662   \n",
       "7036                 0.2097  ...                 0.0647   \n",
       "7037                 0.1578  ...                 0.0062   \n",
       "\n",
       "     gnomad40_genome_AF_amr gnomad40_genome_AF_asj gnomad40_genome_AF_eas  \\\n",
       "0                         .                      .                      .   \n",
       "1                    0.0001                      0                      0   \n",
       "2                    0.0001                      0                      0   \n",
       "3                         0                      0                      0   \n",
       "4                         0                      0                 0.0002   \n",
       "...                     ...                    ...                    ...   \n",
       "7033                      0                      0                      0   \n",
       "7034              6.805e-05                      0                 0.0002   \n",
       "7035                 0.0692                 0.0709                 0.0022   \n",
       "7036                 0.1589                 0.1773                 0.0702   \n",
       "7037                 0.0308                 0.0201                 0.0087   \n",
       "\n",
       "     gnomad40_genome_AF_fin gnomad40_genome_AF_mid gnomad40_genome_AF_nfe  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                      0                      0   \n",
       "2                         0                      0              4.411e-05   \n",
       "3                         0                      0               1.47e-05   \n",
       "4                         0                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "7033                      0                      0              1.484e-05   \n",
       "7034                      0                      0                 0.0001   \n",
       "7035                 0.0349                 0.1206                 0.0602   \n",
       "7036                 0.2510                 0.1939                 0.1599   \n",
       "7037                 0.0598                 0.0287                 0.0198   \n",
       "\n",
       "     gnomad40_genome_AF_remaining gnomad40_genome_AF_sas      anno  \n",
       "0                               .                      .  15_Quies  \n",
       "1                               0                      0  15_Quies  \n",
       "2                          0.0009                 0.0182  15_Quies  \n",
       "3                               0                 0.0008  15_Quies  \n",
       "4                               0                 0.0002  15_Quies  \n",
       "...                           ...                    ...       ...  \n",
       "7033                            0                 0.0002      4_Tx  \n",
       "7034                            0                      0  15_Quies  \n",
       "7035                       0.0888                 0.0348  15_Quies  \n",
       "7036                       0.1744                 0.2595       NaN  \n",
       "7037                       0.0278                 0.0359       NaN  \n",
       "\n",
       "[7036 rows x 28 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos</th>\n",
       "      <th>chr_x</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad40_genome_AF_afr</th>\n",
       "      <th>gnomad40_genome_AF_ami</th>\n",
       "      <th>gnomad40_genome_AF_amr</th>\n",
       "      <th>gnomad40_genome_AF_asj</th>\n",
       "      <th>gnomad40_genome_AF_eas</th>\n",
       "      <th>gnomad40_genome_AF_fin</th>\n",
       "      <th>gnomad40_genome_AF_mid</th>\n",
       "      <th>gnomad40_genome_AF_nfe</th>\n",
       "      <th>gnomad40_genome_AF_remaining</th>\n",
       "      <th>gnomad40_genome_AF_sas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>32434</td>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.018746</td>\n",
       "      <td>165.452320</td>\n",
       "      <td>0.097936</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>47480</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015024</td>\n",
       "      <td>126.894470</td>\n",
       "      <td>0.070111</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>48195</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.019221</td>\n",
       "      <td>0.015570</td>\n",
       "      <td>77.986465</td>\n",
       "      <td>0.043584</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>...</td>\n",
       "      <td>2.407e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.411e-05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.0182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>48202</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.016036</td>\n",
       "      <td>115.160090</td>\n",
       "      <td>0.067440</td>\n",
       "      <td>3.284e-05</td>\n",
       "      <td>3.281e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>60400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>0.015248</td>\n",
       "      <td>79.551820</td>\n",
       "      <td>0.037764</td>\n",
       "      <td>1.314e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7032</th>\n",
       "      <td>197576155</td>\n",
       "      <td>chr3</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015796</td>\n",
       "      <td>94.429090</td>\n",
       "      <td>0.051905</td>\n",
       "      <td>2.74e-05</td>\n",
       "      <td>2.684e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>5.208e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.484e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7033</th>\n",
       "      <td>197636492</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015411</td>\n",
       "      <td>62.338760</td>\n",
       "      <td>0.026065</td>\n",
       "      <td>8.088e-05</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>...</td>\n",
       "      <td>5.003e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>6.805e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7034</th>\n",
       "      <td>197656793</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.055652</td>\n",
       "      <td>0.017351</td>\n",
       "      <td>84.323980</td>\n",
       "      <td>0.037448</td>\n",
       "      <td>0.1119</td>\n",
       "      <td>0.1683</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3134</td>\n",
       "      <td>0.1662</td>\n",
       "      <td>0.0692</td>\n",
       "      <td>0.0709</td>\n",
       "      <td>0.0022</td>\n",
       "      <td>0.0349</td>\n",
       "      <td>0.1206</td>\n",
       "      <td>0.0602</td>\n",
       "      <td>0.0888</td>\n",
       "      <td>0.0348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7035</th>\n",
       "      <td>198116021</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.241652</td>\n",
       "      <td>0.018788</td>\n",
       "      <td>87.795296</td>\n",
       "      <td>0.039123</td>\n",
       "      <td>0.2093</td>\n",
       "      <td>0.2097</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.0647</td>\n",
       "      <td>0.1589</td>\n",
       "      <td>0.1773</td>\n",
       "      <td>0.0702</td>\n",
       "      <td>0.2510</td>\n",
       "      <td>0.1939</td>\n",
       "      <td>0.1599</td>\n",
       "      <td>0.1744</td>\n",
       "      <td>0.2595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7036</th>\n",
       "      <td>198117279</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.016466</td>\n",
       "      <td>106.986080</td>\n",
       "      <td>0.056975</td>\n",
       "      <td>0.0301</td>\n",
       "      <td>0.1578</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0456</td>\n",
       "      <td>0.0062</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.0201</td>\n",
       "      <td>0.0087</td>\n",
       "      <td>0.0598</td>\n",
       "      <td>0.0287</td>\n",
       "      <td>0.0198</td>\n",
       "      <td>0.0278</td>\n",
       "      <td>0.0359</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7037 rows Ã— 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pos chr_x ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0         32434  chr3     A     C     0.000261         0.018746   \n",
       "1         47480  chr3     T     C     0.000174         0.015024   \n",
       "2         48195  chr3     T     C     0.019221         0.015570   \n",
       "3         48202  chr3     G     C     0.001739         0.016036   \n",
       "4         60400  chr3     G     C     0.001044         0.015248   \n",
       "...         ...   ...   ...   ...          ...              ...   \n",
       "7032  197576155  chr3     C     G     0.000174         0.015796   \n",
       "7033  197636492  chr3     T     C     0.000174         0.015411   \n",
       "7034  197656793  chr3     T     C     0.055652         0.017351   \n",
       "7035  198116021  chr3     T     C     0.241652         0.018788   \n",
       "7036  198117279  chr3     T     C     0.000177         0.016466   \n",
       "\n",
       "      log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                 165.452320        0.097936                  .   \n",
       "1                 126.894470        0.070111             0.0001   \n",
       "2                  77.986465        0.043584             0.0006   \n",
       "3                 115.160090        0.067440          3.284e-05   \n",
       "4                  79.551820        0.037764          1.314e-05   \n",
       "...                      ...             ...                ...   \n",
       "7032               94.429090        0.051905           2.74e-05   \n",
       "7033               62.338760        0.026065          8.088e-05   \n",
       "7034               84.323980        0.037448             0.1119   \n",
       "7035               87.795296        0.039123             0.2093   \n",
       "7036              106.986080        0.056975             0.0301   \n",
       "\n",
       "     gnomad40_genome_AF_raw  ... gnomad40_genome_AF_afr  \\\n",
       "0                         .  ...                      .   \n",
       "1                    0.0001  ...                 0.0004   \n",
       "2                    0.0006  ...              2.407e-05   \n",
       "3                 3.281e-05  ...                      0   \n",
       "4                 1.312e-05  ...                      0   \n",
       "...                     ...  ...                    ...   \n",
       "7032              2.684e-05  ...              5.208e-05   \n",
       "7033                 0.0011  ...              5.003e-05   \n",
       "7034                 0.1683  ...                 0.3134   \n",
       "7035                 0.2097  ...                 0.3173   \n",
       "7036                 0.1578  ...                 0.0456   \n",
       "\n",
       "     gnomad40_genome_AF_ami gnomad40_genome_AF_amr gnomad40_genome_AF_asj  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                 0.0001                      0   \n",
       "2                         0                 0.0001                      0   \n",
       "3                         0                      0                      0   \n",
       "4                         0                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "7032                      0                      0                      0   \n",
       "7033                      0              6.805e-05                      0   \n",
       "7034                 0.1662                 0.0692                 0.0709   \n",
       "7035                 0.0647                 0.1589                 0.1773   \n",
       "7036                 0.0062                 0.0308                 0.0201   \n",
       "\n",
       "     gnomad40_genome_AF_eas gnomad40_genome_AF_fin gnomad40_genome_AF_mid  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                      0                      0   \n",
       "2                         0                      0                      0   \n",
       "3                         0                      0                      0   \n",
       "4                    0.0002                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "7032                      0                      0                      0   \n",
       "7033                 0.0002                      0                      0   \n",
       "7034                 0.0022                 0.0349                 0.1206   \n",
       "7035                 0.0702                 0.2510                 0.1939   \n",
       "7036                 0.0087                 0.0598                 0.0287   \n",
       "\n",
       "     gnomad40_genome_AF_nfe gnomad40_genome_AF_remaining  \\\n",
       "0                         .                            .   \n",
       "1                         0                            0   \n",
       "2                 4.411e-05                       0.0009   \n",
       "3                  1.47e-05                            0   \n",
       "4                         0                            0   \n",
       "...                     ...                          ...   \n",
       "7032              1.484e-05                            0   \n",
       "7033                 0.0001                            0   \n",
       "7034                 0.0602                       0.0888   \n",
       "7035                 0.1599                       0.1744   \n",
       "7036                 0.0198                       0.0278   \n",
       "\n",
       "     gnomad40_genome_AF_sas  \n",
       "0                         .  \n",
       "1                         0  \n",
       "2                    0.0182  \n",
       "3                    0.0008  \n",
       "4                    0.0002  \n",
       "...                     ...  \n",
       "7032                 0.0002  \n",
       "7033                      0  \n",
       "7034                 0.0348  \n",
       "7035                 0.2595  \n",
       "7036                 0.0359  \n",
       "\n",
       "[7037 rows x 27 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_data_old=snp_data_new.drop(columns=['anno'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos</th>\n",
       "      <th>chr_x</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad40_genome_AF_afr</th>\n",
       "      <th>gnomad40_genome_AF_ami</th>\n",
       "      <th>gnomad40_genome_AF_amr</th>\n",
       "      <th>gnomad40_genome_AF_asj</th>\n",
       "      <th>gnomad40_genome_AF_eas</th>\n",
       "      <th>gnomad40_genome_AF_fin</th>\n",
       "      <th>gnomad40_genome_AF_mid</th>\n",
       "      <th>gnomad40_genome_AF_nfe</th>\n",
       "      <th>gnomad40_genome_AF_remaining</th>\n",
       "      <th>gnomad40_genome_AF_sas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>32434</td>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.018746</td>\n",
       "      <td>165.452320</td>\n",
       "      <td>0.097936</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>47480</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015024</td>\n",
       "      <td>126.894470</td>\n",
       "      <td>0.070111</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>48195</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.019221</td>\n",
       "      <td>0.015570</td>\n",
       "      <td>77.986465</td>\n",
       "      <td>0.043584</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>...</td>\n",
       "      <td>2.407e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.411e-05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.0182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>48202</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.016036</td>\n",
       "      <td>115.160090</td>\n",
       "      <td>0.067440</td>\n",
       "      <td>3.284e-05</td>\n",
       "      <td>3.281e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>60400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>0.015248</td>\n",
       "      <td>79.551820</td>\n",
       "      <td>0.037764</td>\n",
       "      <td>1.314e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7033</th>\n",
       "      <td>197576155</td>\n",
       "      <td>chr3</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015796</td>\n",
       "      <td>94.429090</td>\n",
       "      <td>0.051905</td>\n",
       "      <td>2.74e-05</td>\n",
       "      <td>2.684e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>5.208e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.484e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7034</th>\n",
       "      <td>197636492</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015411</td>\n",
       "      <td>62.338760</td>\n",
       "      <td>0.026065</td>\n",
       "      <td>8.088e-05</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>...</td>\n",
       "      <td>5.003e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>6.805e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7035</th>\n",
       "      <td>197656793</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.055652</td>\n",
       "      <td>0.017351</td>\n",
       "      <td>84.323980</td>\n",
       "      <td>0.037448</td>\n",
       "      <td>0.1119</td>\n",
       "      <td>0.1683</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3134</td>\n",
       "      <td>0.1662</td>\n",
       "      <td>0.0692</td>\n",
       "      <td>0.0709</td>\n",
       "      <td>0.0022</td>\n",
       "      <td>0.0349</td>\n",
       "      <td>0.1206</td>\n",
       "      <td>0.0602</td>\n",
       "      <td>0.0888</td>\n",
       "      <td>0.0348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7036</th>\n",
       "      <td>198116021</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.241652</td>\n",
       "      <td>0.018788</td>\n",
       "      <td>87.795296</td>\n",
       "      <td>0.039123</td>\n",
       "      <td>0.2093</td>\n",
       "      <td>0.2097</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.0647</td>\n",
       "      <td>0.1589</td>\n",
       "      <td>0.1773</td>\n",
       "      <td>0.0702</td>\n",
       "      <td>0.2510</td>\n",
       "      <td>0.1939</td>\n",
       "      <td>0.1599</td>\n",
       "      <td>0.1744</td>\n",
       "      <td>0.2595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7037</th>\n",
       "      <td>198117279</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.016466</td>\n",
       "      <td>106.986080</td>\n",
       "      <td>0.056975</td>\n",
       "      <td>0.0301</td>\n",
       "      <td>0.1578</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0456</td>\n",
       "      <td>0.0062</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.0201</td>\n",
       "      <td>0.0087</td>\n",
       "      <td>0.0598</td>\n",
       "      <td>0.0287</td>\n",
       "      <td>0.0198</td>\n",
       "      <td>0.0278</td>\n",
       "      <td>0.0359</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7038 rows Ã— 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pos chr_x ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0         32434  chr3     A     C     0.000261         0.018746   \n",
       "1         47480  chr3     T     C     0.000174         0.015024   \n",
       "2         48195  chr3     T     C     0.019221         0.015570   \n",
       "3         48202  chr3     G     C     0.001739         0.016036   \n",
       "4         60400  chr3     G     C     0.001044         0.015248   \n",
       "...         ...   ...   ...   ...          ...              ...   \n",
       "7033  197576155  chr3     C     G     0.000174         0.015796   \n",
       "7034  197636492  chr3     T     C     0.000174         0.015411   \n",
       "7035  197656793  chr3     T     C     0.055652         0.017351   \n",
       "7036  198116021  chr3     T     C     0.241652         0.018788   \n",
       "7037  198117279  chr3     T     C     0.000177         0.016466   \n",
       "\n",
       "      log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                 165.452320        0.097936                  .   \n",
       "1                 126.894470        0.070111             0.0001   \n",
       "2                  77.986465        0.043584             0.0006   \n",
       "3                 115.160090        0.067440          3.284e-05   \n",
       "4                  79.551820        0.037764          1.314e-05   \n",
       "...                      ...             ...                ...   \n",
       "7033               94.429090        0.051905           2.74e-05   \n",
       "7034               62.338760        0.026065          8.088e-05   \n",
       "7035               84.323980        0.037448             0.1119   \n",
       "7036               87.795296        0.039123             0.2093   \n",
       "7037              106.986080        0.056975             0.0301   \n",
       "\n",
       "     gnomad40_genome_AF_raw  ... gnomad40_genome_AF_afr  \\\n",
       "0                         .  ...                      .   \n",
       "1                    0.0001  ...                 0.0004   \n",
       "2                    0.0006  ...              2.407e-05   \n",
       "3                 3.281e-05  ...                      0   \n",
       "4                 1.312e-05  ...                      0   \n",
       "...                     ...  ...                    ...   \n",
       "7033              2.684e-05  ...              5.208e-05   \n",
       "7034                 0.0011  ...              5.003e-05   \n",
       "7035                 0.1683  ...                 0.3134   \n",
       "7036                 0.2097  ...                 0.3173   \n",
       "7037                 0.1578  ...                 0.0456   \n",
       "\n",
       "     gnomad40_genome_AF_ami gnomad40_genome_AF_amr gnomad40_genome_AF_asj  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                 0.0001                      0   \n",
       "2                         0                 0.0001                      0   \n",
       "3                         0                      0                      0   \n",
       "4                         0                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "7033                      0                      0                      0   \n",
       "7034                      0              6.805e-05                      0   \n",
       "7035                 0.1662                 0.0692                 0.0709   \n",
       "7036                 0.0647                 0.1589                 0.1773   \n",
       "7037                 0.0062                 0.0308                 0.0201   \n",
       "\n",
       "     gnomad40_genome_AF_eas gnomad40_genome_AF_fin gnomad40_genome_AF_mid  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                      0                      0   \n",
       "2                         0                      0                      0   \n",
       "3                         0                      0                      0   \n",
       "4                    0.0002                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "7033                      0                      0                      0   \n",
       "7034                 0.0002                      0                      0   \n",
       "7035                 0.0022                 0.0349                 0.1206   \n",
       "7036                 0.0702                 0.2510                 0.1939   \n",
       "7037                 0.0087                 0.0598                 0.0287   \n",
       "\n",
       "     gnomad40_genome_AF_nfe gnomad40_genome_AF_remaining  \\\n",
       "0                         .                            .   \n",
       "1                         0                            0   \n",
       "2                 4.411e-05                       0.0009   \n",
       "3                  1.47e-05                            0   \n",
       "4                         0                            0   \n",
       "...                     ...                          ...   \n",
       "7033              1.484e-05                            0   \n",
       "7034                 0.0001                            0   \n",
       "7035                 0.0602                       0.0888   \n",
       "7036                 0.1599                       0.1744   \n",
       "7037                 0.0198                       0.0278   \n",
       "\n",
       "     gnomad40_genome_AF_sas  \n",
       "0                         .  \n",
       "1                         0  \n",
       "2                    0.0182  \n",
       "3                    0.0008  \n",
       "4                    0.0002  \n",
       "...                     ...  \n",
       "7033                 0.0002  \n",
       "7034                      0  \n",
       "7035                 0.0348  \n",
       "7036                 0.2595  \n",
       "7037                 0.0359  \n",
       "\n",
       "[7038 rows x 27 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data_old"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "directory3='/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model1/final_data/'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos</th>\n",
       "      <th>chr_x</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad40_genome_AF_afr</th>\n",
       "      <th>gnomad40_genome_AF_ami</th>\n",
       "      <th>gnomad40_genome_AF_amr</th>\n",
       "      <th>gnomad40_genome_AF_asj</th>\n",
       "      <th>gnomad40_genome_AF_eas</th>\n",
       "      <th>gnomad40_genome_AF_fin</th>\n",
       "      <th>gnomad40_genome_AF_mid</th>\n",
       "      <th>gnomad40_genome_AF_nfe</th>\n",
       "      <th>gnomad40_genome_AF_remaining</th>\n",
       "      <th>gnomad40_genome_AF_sas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>32434</td>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.018746</td>\n",
       "      <td>165.452320</td>\n",
       "      <td>0.097936</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>47480</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015024</td>\n",
       "      <td>126.894470</td>\n",
       "      <td>0.070111</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>48195</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.019221</td>\n",
       "      <td>0.015570</td>\n",
       "      <td>77.986465</td>\n",
       "      <td>0.043584</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>...</td>\n",
       "      <td>2.407e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.411e-05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.0182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>48202</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.016036</td>\n",
       "      <td>115.160090</td>\n",
       "      <td>0.067440</td>\n",
       "      <td>3.284e-05</td>\n",
       "      <td>3.281e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>60400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>0.015248</td>\n",
       "      <td>79.551820</td>\n",
       "      <td>0.037764</td>\n",
       "      <td>1.314e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5712</th>\n",
       "      <td>157058129</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000609</td>\n",
       "      <td>0.015688</td>\n",
       "      <td>51.360200</td>\n",
       "      <td>0.021876</td>\n",
       "      <td>1.313e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5713</th>\n",
       "      <td>157138658</td>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.012261</td>\n",
       "      <td>0.015064</td>\n",
       "      <td>56.561960</td>\n",
       "      <td>0.032618</td>\n",
       "      <td>0.0737</td>\n",
       "      <td>0.0738</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2398</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0302</td>\n",
       "      <td>0.0184</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>0.0680</td>\n",
       "      <td>0.0072</td>\n",
       "      <td>0.0592</td>\n",
       "      <td>0.0180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5714</th>\n",
       "      <td>157141400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.092956</td>\n",
       "      <td>0.017347</td>\n",
       "      <td>78.606330</td>\n",
       "      <td>0.039950</td>\n",
       "      <td>0.0414</td>\n",
       "      <td>0.0415</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0104</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0724</td>\n",
       "      <td>0.0357</td>\n",
       "      <td>0.0839</td>\n",
       "      <td>0.0811</td>\n",
       "      <td>0.0272</td>\n",
       "      <td>0.0421</td>\n",
       "      <td>0.0255</td>\n",
       "      <td>0.0845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5715</th>\n",
       "      <td>157145900</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.019555</td>\n",
       "      <td>132.004640</td>\n",
       "      <td>0.078494</td>\n",
       "      <td>0.0225</td>\n",
       "      <td>0.0225</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0779</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0080</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0.0199</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5716</th>\n",
       "      <td>157161336</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015292</td>\n",
       "      <td>75.022800</td>\n",
       "      <td>0.036740</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5717 rows Ã— 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pos chr_x ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0         32434  chr3     A     C     0.000261         0.018746   \n",
       "1         47480  chr3     T     C     0.000174         0.015024   \n",
       "2         48195  chr3     T     C     0.019221         0.015570   \n",
       "3         48202  chr3     G     C     0.001739         0.016036   \n",
       "4         60400  chr3     G     C     0.001044         0.015248   \n",
       "...         ...   ...   ...   ...          ...              ...   \n",
       "5712  157058129  chr3     T     G     0.000609         0.015688   \n",
       "5713  157138658  chr3     A     G     0.012261         0.015064   \n",
       "5714  157141400  chr3     T     C     0.092956         0.017347   \n",
       "5715  157145900  chr3     T     C     0.000261         0.019555   \n",
       "5716  157161336  chr3     G     C     0.000174         0.015292   \n",
       "\n",
       "      log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                 165.452320        0.097936                  .   \n",
       "1                 126.894470        0.070111             0.0001   \n",
       "2                  77.986465        0.043584             0.0006   \n",
       "3                 115.160090        0.067440          3.284e-05   \n",
       "4                  79.551820        0.037764          1.314e-05   \n",
       "...                      ...             ...                ...   \n",
       "5712               51.360200        0.021876          1.313e-05   \n",
       "5713               56.561960        0.032618             0.0737   \n",
       "5714               78.606330        0.039950             0.0414   \n",
       "5715              132.004640        0.078494             0.0225   \n",
       "5716               75.022800        0.036740                  .   \n",
       "\n",
       "     gnomad40_genome_AF_raw  ... gnomad40_genome_AF_afr  \\\n",
       "0                         .  ...                      .   \n",
       "1                    0.0001  ...                 0.0004   \n",
       "2                    0.0006  ...              2.407e-05   \n",
       "3                 3.281e-05  ...                      0   \n",
       "4                 1.312e-05  ...                      0   \n",
       "...                     ...  ...                    ...   \n",
       "5712              1.312e-05  ...                      0   \n",
       "5713                 0.0738  ...                 0.2398   \n",
       "5714                 0.0415  ...                 0.0104   \n",
       "5715                 0.0225  ...                 0.0779   \n",
       "5716                      .  ...                      .   \n",
       "\n",
       "     gnomad40_genome_AF_ami gnomad40_genome_AF_amr gnomad40_genome_AF_asj  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                 0.0001                      0   \n",
       "2                         0                 0.0001                      0   \n",
       "3                         0                      0                      0   \n",
       "4                         0                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "5712                      0                      0                      0   \n",
       "5713                      0                 0.0302                 0.0184   \n",
       "5714                 0.0066                 0.0724                 0.0357   \n",
       "5715                      0                 0.0080                      0   \n",
       "5716                      .                      .                      .   \n",
       "\n",
       "     gnomad40_genome_AF_eas gnomad40_genome_AF_fin gnomad40_genome_AF_mid  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                      0                      0   \n",
       "2                         0                      0                      0   \n",
       "3                         0                      0                      0   \n",
       "4                    0.0002                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "5712                      0                      0                      0   \n",
       "5713                 0.0002                 0.0013                 0.0680   \n",
       "5714                 0.0839                 0.0811                 0.0272   \n",
       "5715                      0                 0.0003                 0.0034   \n",
       "5716                      .                      .                      .   \n",
       "\n",
       "     gnomad40_genome_AF_nfe gnomad40_genome_AF_remaining  \\\n",
       "0                         .                            .   \n",
       "1                         0                            0   \n",
       "2                 4.411e-05                       0.0009   \n",
       "3                  1.47e-05                            0   \n",
       "4                         0                            0   \n",
       "...                     ...                          ...   \n",
       "5712                      0                            0   \n",
       "5713                 0.0072                       0.0592   \n",
       "5714                 0.0421                       0.0255   \n",
       "5715                 0.0003                       0.0199   \n",
       "5716                      .                            .   \n",
       "\n",
       "     gnomad40_genome_AF_sas  \n",
       "0                         .  \n",
       "1                         0  \n",
       "2                    0.0182  \n",
       "3                    0.0008  \n",
       "4                    0.0002  \n",
       "...                     ...  \n",
       "5712                 0.0004  \n",
       "5713                 0.0180  \n",
       "5714                 0.0845  \n",
       "5715                 0.0002  \n",
       "5716                      .  \n",
       "\n",
       "[5717 rows x 27 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data.head(5717)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos</th>\n",
       "      <th>chr_x</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad40_genome_AF_afr</th>\n",
       "      <th>gnomad40_genome_AF_ami</th>\n",
       "      <th>gnomad40_genome_AF_amr</th>\n",
       "      <th>gnomad40_genome_AF_asj</th>\n",
       "      <th>gnomad40_genome_AF_eas</th>\n",
       "      <th>gnomad40_genome_AF_fin</th>\n",
       "      <th>gnomad40_genome_AF_mid</th>\n",
       "      <th>gnomad40_genome_AF_nfe</th>\n",
       "      <th>gnomad40_genome_AF_remaining</th>\n",
       "      <th>gnomad40_genome_AF_sas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>32434</td>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.018746</td>\n",
       "      <td>165.452320</td>\n",
       "      <td>0.097936</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>47480</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015024</td>\n",
       "      <td>126.894470</td>\n",
       "      <td>0.070111</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>48195</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.019221</td>\n",
       "      <td>0.015570</td>\n",
       "      <td>77.986465</td>\n",
       "      <td>0.043584</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>...</td>\n",
       "      <td>2.407e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.411e-05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.0182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>48202</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.016036</td>\n",
       "      <td>115.160090</td>\n",
       "      <td>0.067440</td>\n",
       "      <td>3.284e-05</td>\n",
       "      <td>3.281e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>60400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>0.015248</td>\n",
       "      <td>79.551820</td>\n",
       "      <td>0.037764</td>\n",
       "      <td>1.314e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5712</th>\n",
       "      <td>157058129</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000609</td>\n",
       "      <td>0.015688</td>\n",
       "      <td>51.360200</td>\n",
       "      <td>0.021876</td>\n",
       "      <td>1.313e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5713</th>\n",
       "      <td>157138658</td>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.012261</td>\n",
       "      <td>0.015064</td>\n",
       "      <td>56.561960</td>\n",
       "      <td>0.032618</td>\n",
       "      <td>0.0737</td>\n",
       "      <td>0.0738</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2398</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0302</td>\n",
       "      <td>0.0184</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>0.0680</td>\n",
       "      <td>0.0072</td>\n",
       "      <td>0.0592</td>\n",
       "      <td>0.0180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5714</th>\n",
       "      <td>157141400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.092956</td>\n",
       "      <td>0.017347</td>\n",
       "      <td>78.606330</td>\n",
       "      <td>0.039950</td>\n",
       "      <td>0.0414</td>\n",
       "      <td>0.0415</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0104</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0724</td>\n",
       "      <td>0.0357</td>\n",
       "      <td>0.0839</td>\n",
       "      <td>0.0811</td>\n",
       "      <td>0.0272</td>\n",
       "      <td>0.0421</td>\n",
       "      <td>0.0255</td>\n",
       "      <td>0.0845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5715</th>\n",
       "      <td>157141400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.092956</td>\n",
       "      <td>0.017347</td>\n",
       "      <td>78.606330</td>\n",
       "      <td>0.039950</td>\n",
       "      <td>0.0414</td>\n",
       "      <td>0.0415</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0104</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0724</td>\n",
       "      <td>0.0357</td>\n",
       "      <td>0.0839</td>\n",
       "      <td>0.0811</td>\n",
       "      <td>0.0272</td>\n",
       "      <td>0.0421</td>\n",
       "      <td>0.0255</td>\n",
       "      <td>0.0845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5716</th>\n",
       "      <td>157145900</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.019555</td>\n",
       "      <td>132.004640</td>\n",
       "      <td>0.078494</td>\n",
       "      <td>0.0225</td>\n",
       "      <td>0.0225</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0779</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0080</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0.0199</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5717 rows Ã— 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pos chr_x ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0         32434  chr3     A     C     0.000261         0.018746   \n",
       "1         47480  chr3     T     C     0.000174         0.015024   \n",
       "2         48195  chr3     T     C     0.019221         0.015570   \n",
       "3         48202  chr3     G     C     0.001739         0.016036   \n",
       "4         60400  chr3     G     C     0.001044         0.015248   \n",
       "...         ...   ...   ...   ...          ...              ...   \n",
       "5712  157058129  chr3     T     G     0.000609         0.015688   \n",
       "5713  157138658  chr3     A     G     0.012261         0.015064   \n",
       "5714  157141400  chr3     T     C     0.092956         0.017347   \n",
       "5715  157141400  chr3     T     C     0.092956         0.017347   \n",
       "5716  157145900  chr3     T     C     0.000261         0.019555   \n",
       "\n",
       "      log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                 165.452320        0.097936                  .   \n",
       "1                 126.894470        0.070111             0.0001   \n",
       "2                  77.986465        0.043584             0.0006   \n",
       "3                 115.160090        0.067440          3.284e-05   \n",
       "4                  79.551820        0.037764          1.314e-05   \n",
       "...                      ...             ...                ...   \n",
       "5712               51.360200        0.021876          1.313e-05   \n",
       "5713               56.561960        0.032618             0.0737   \n",
       "5714               78.606330        0.039950             0.0414   \n",
       "5715               78.606330        0.039950             0.0414   \n",
       "5716              132.004640        0.078494             0.0225   \n",
       "\n",
       "     gnomad40_genome_AF_raw  ... gnomad40_genome_AF_afr  \\\n",
       "0                         .  ...                      .   \n",
       "1                    0.0001  ...                 0.0004   \n",
       "2                    0.0006  ...              2.407e-05   \n",
       "3                 3.281e-05  ...                      0   \n",
       "4                 1.312e-05  ...                      0   \n",
       "...                     ...  ...                    ...   \n",
       "5712              1.312e-05  ...                      0   \n",
       "5713                 0.0738  ...                 0.2398   \n",
       "5714                 0.0415  ...                 0.0104   \n",
       "5715                 0.0415  ...                 0.0104   \n",
       "5716                 0.0225  ...                 0.0779   \n",
       "\n",
       "     gnomad40_genome_AF_ami gnomad40_genome_AF_amr gnomad40_genome_AF_asj  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                 0.0001                      0   \n",
       "2                         0                 0.0001                      0   \n",
       "3                         0                      0                      0   \n",
       "4                         0                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "5712                      0                      0                      0   \n",
       "5713                      0                 0.0302                 0.0184   \n",
       "5714                 0.0066                 0.0724                 0.0357   \n",
       "5715                 0.0066                 0.0724                 0.0357   \n",
       "5716                      0                 0.0080                      0   \n",
       "\n",
       "     gnomad40_genome_AF_eas gnomad40_genome_AF_fin gnomad40_genome_AF_mid  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                      0                      0   \n",
       "2                         0                      0                      0   \n",
       "3                         0                      0                      0   \n",
       "4                    0.0002                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "5712                      0                      0                      0   \n",
       "5713                 0.0002                 0.0013                 0.0680   \n",
       "5714                 0.0839                 0.0811                 0.0272   \n",
       "5715                 0.0839                 0.0811                 0.0272   \n",
       "5716                      0                 0.0003                 0.0034   \n",
       "\n",
       "     gnomad40_genome_AF_nfe gnomad40_genome_AF_remaining  \\\n",
       "0                         .                            .   \n",
       "1                         0                            0   \n",
       "2                 4.411e-05                       0.0009   \n",
       "3                  1.47e-05                            0   \n",
       "4                         0                            0   \n",
       "...                     ...                          ...   \n",
       "5712                      0                            0   \n",
       "5713                 0.0072                       0.0592   \n",
       "5714                 0.0421                       0.0255   \n",
       "5715                 0.0421                       0.0255   \n",
       "5716                 0.0003                       0.0199   \n",
       "\n",
       "     gnomad40_genome_AF_sas  \n",
       "0                         .  \n",
       "1                         0  \n",
       "2                    0.0182  \n",
       "3                    0.0008  \n",
       "4                    0.0002  \n",
       "...                     ...  \n",
       "5712                 0.0004  \n",
       "5713                 0.0180  \n",
       "5714                 0.0845  \n",
       "5715                 0.0845  \n",
       "5716                 0.0002  \n",
       "\n",
       "[5717 rows x 27 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data_old.head(5717)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos</th>\n",
       "      <th>chr_x</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad40_genome_AF_ami</th>\n",
       "      <th>gnomad40_genome_AF_amr</th>\n",
       "      <th>gnomad40_genome_AF_asj</th>\n",
       "      <th>gnomad40_genome_AF_eas</th>\n",
       "      <th>gnomad40_genome_AF_fin</th>\n",
       "      <th>gnomad40_genome_AF_mid</th>\n",
       "      <th>gnomad40_genome_AF_nfe</th>\n",
       "      <th>gnomad40_genome_AF_remaining</th>\n",
       "      <th>gnomad40_genome_AF_sas</th>\n",
       "      <th>anno</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>32434</td>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.018746</td>\n",
       "      <td>165.452320</td>\n",
       "      <td>0.097936</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>47480</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015024</td>\n",
       "      <td>126.894470</td>\n",
       "      <td>0.070111</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>48195</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.019221</td>\n",
       "      <td>0.015570</td>\n",
       "      <td>77.986465</td>\n",
       "      <td>0.043584</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.411e-05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.0182</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>48202</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.016036</td>\n",
       "      <td>115.160090</td>\n",
       "      <td>0.067440</td>\n",
       "      <td>3.284e-05</td>\n",
       "      <td>3.281e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0008</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>60400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>0.015248</td>\n",
       "      <td>79.551820</td>\n",
       "      <td>0.037764</td>\n",
       "      <td>1.314e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5712</th>\n",
       "      <td>157058129</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000609</td>\n",
       "      <td>0.015688</td>\n",
       "      <td>51.360200</td>\n",
       "      <td>0.021876</td>\n",
       "      <td>1.313e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5713</th>\n",
       "      <td>157138658</td>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.012261</td>\n",
       "      <td>0.015064</td>\n",
       "      <td>56.561960</td>\n",
       "      <td>0.032618</td>\n",
       "      <td>0.0737</td>\n",
       "      <td>0.0738</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0302</td>\n",
       "      <td>0.0184</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>0.0680</td>\n",
       "      <td>0.0072</td>\n",
       "      <td>0.0592</td>\n",
       "      <td>0.0180</td>\n",
       "      <td>5_TxWk</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5714</th>\n",
       "      <td>157141400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.092956</td>\n",
       "      <td>0.017347</td>\n",
       "      <td>78.606330</td>\n",
       "      <td>0.039950</td>\n",
       "      <td>0.0414</td>\n",
       "      <td>0.0415</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0724</td>\n",
       "      <td>0.0357</td>\n",
       "      <td>0.0839</td>\n",
       "      <td>0.0811</td>\n",
       "      <td>0.0272</td>\n",
       "      <td>0.0421</td>\n",
       "      <td>0.0255</td>\n",
       "      <td>0.0845</td>\n",
       "      <td>7_Enh</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5715</th>\n",
       "      <td>157141400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.092956</td>\n",
       "      <td>0.017347</td>\n",
       "      <td>78.606330</td>\n",
       "      <td>0.039950</td>\n",
       "      <td>0.0414</td>\n",
       "      <td>0.0415</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0724</td>\n",
       "      <td>0.0357</td>\n",
       "      <td>0.0839</td>\n",
       "      <td>0.0811</td>\n",
       "      <td>0.0272</td>\n",
       "      <td>0.0421</td>\n",
       "      <td>0.0255</td>\n",
       "      <td>0.0845</td>\n",
       "      <td>5_TxWk</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5716</th>\n",
       "      <td>157145900</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.019555</td>\n",
       "      <td>132.004640</td>\n",
       "      <td>0.078494</td>\n",
       "      <td>0.0225</td>\n",
       "      <td>0.0225</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0080</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0.0199</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>5_TxWk</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5717 rows Ã— 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pos chr_x ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0         32434  chr3     A     C     0.000261         0.018746   \n",
       "1         47480  chr3     T     C     0.000174         0.015024   \n",
       "2         48195  chr3     T     C     0.019221         0.015570   \n",
       "3         48202  chr3     G     C     0.001739         0.016036   \n",
       "4         60400  chr3     G     C     0.001044         0.015248   \n",
       "...         ...   ...   ...   ...          ...              ...   \n",
       "5712  157058129  chr3     T     G     0.000609         0.015688   \n",
       "5713  157138658  chr3     A     G     0.012261         0.015064   \n",
       "5714  157141400  chr3     T     C     0.092956         0.017347   \n",
       "5715  157141400  chr3     T     C     0.092956         0.017347   \n",
       "5716  157145900  chr3     T     C     0.000261         0.019555   \n",
       "\n",
       "      log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                 165.452320        0.097936                  .   \n",
       "1                 126.894470        0.070111             0.0001   \n",
       "2                  77.986465        0.043584             0.0006   \n",
       "3                 115.160090        0.067440          3.284e-05   \n",
       "4                  79.551820        0.037764          1.314e-05   \n",
       "...                      ...             ...                ...   \n",
       "5712               51.360200        0.021876          1.313e-05   \n",
       "5713               56.561960        0.032618             0.0737   \n",
       "5714               78.606330        0.039950             0.0414   \n",
       "5715               78.606330        0.039950             0.0414   \n",
       "5716              132.004640        0.078494             0.0225   \n",
       "\n",
       "     gnomad40_genome_AF_raw  ... gnomad40_genome_AF_ami  \\\n",
       "0                         .  ...                      .   \n",
       "1                    0.0001  ...                      0   \n",
       "2                    0.0006  ...                      0   \n",
       "3                 3.281e-05  ...                      0   \n",
       "4                 1.312e-05  ...                      0   \n",
       "...                     ...  ...                    ...   \n",
       "5712              1.312e-05  ...                      0   \n",
       "5713                 0.0738  ...                      0   \n",
       "5714                 0.0415  ...                 0.0066   \n",
       "5715                 0.0415  ...                 0.0066   \n",
       "5716                 0.0225  ...                      0   \n",
       "\n",
       "     gnomad40_genome_AF_amr gnomad40_genome_AF_asj gnomad40_genome_AF_eas  \\\n",
       "0                         .                      .                      .   \n",
       "1                    0.0001                      0                      0   \n",
       "2                    0.0001                      0                      0   \n",
       "3                         0                      0                      0   \n",
       "4                         0                      0                 0.0002   \n",
       "...                     ...                    ...                    ...   \n",
       "5712                      0                      0                      0   \n",
       "5713                 0.0302                 0.0184                 0.0002   \n",
       "5714                 0.0724                 0.0357                 0.0839   \n",
       "5715                 0.0724                 0.0357                 0.0839   \n",
       "5716                 0.0080                      0                      0   \n",
       "\n",
       "     gnomad40_genome_AF_fin gnomad40_genome_AF_mid gnomad40_genome_AF_nfe  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                      0                      0   \n",
       "2                         0                      0              4.411e-05   \n",
       "3                         0                      0               1.47e-05   \n",
       "4                         0                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "5712                      0                      0                      0   \n",
       "5713                 0.0013                 0.0680                 0.0072   \n",
       "5714                 0.0811                 0.0272                 0.0421   \n",
       "5715                 0.0811                 0.0272                 0.0421   \n",
       "5716                 0.0003                 0.0034                 0.0003   \n",
       "\n",
       "     gnomad40_genome_AF_remaining gnomad40_genome_AF_sas      anno  \n",
       "0                               .                      .  15_Quies  \n",
       "1                               0                      0  15_Quies  \n",
       "2                          0.0009                 0.0182  15_Quies  \n",
       "3                               0                 0.0008  15_Quies  \n",
       "4                               0                 0.0002  15_Quies  \n",
       "...                           ...                    ...       ...  \n",
       "5712                            0                 0.0004  15_Quies  \n",
       "5713                       0.0592                 0.0180    5_TxWk  \n",
       "5714                       0.0255                 0.0845     7_Enh  \n",
       "5715                       0.0255                 0.0845    5_TxWk  \n",
       "5716                       0.0199                 0.0002    5_TxWk  \n",
       "\n",
       "[5717 rows x 28 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data_new.head(5717)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty DataFrame\n",
      "Columns: [pos, chr_x, ref_x, alt_x, allele_freq, log_counts_diff, log_probs_diff_abs_sum, probs_jsd_diff, gnomad40_genome_AF, gnomad40_genome_AF_raw, gnomad40_genome_AF_XX, gnomad40_genome_AF_XY, gnomad40_genome_AF_grpmax, gnomad40_genome_faf95, gnomad40_genome_faf99, gnomad40_genome_fafmax_faf95_max, gnomad40_genome_fafmax_faf99_max, gnomad40_genome_AF_afr, gnomad40_genome_AF_ami, gnomad40_genome_AF_amr, gnomad40_genome_AF_asj, gnomad40_genome_AF_eas, gnomad40_genome_AF_fin, gnomad40_genome_AF_mid, gnomad40_genome_AF_nfe, gnomad40_genome_AF_remaining, gnomad40_genome_AF_sas]\n",
      "Index: []\n",
      "\n",
      "[0 rows x 27 columns]\n"
     ]
    }
   ],
   "source": [
    "# Merge the dataframes with indicator=True\n",
    "merged = pd.merge(snp_data_old, snp_data, how='outer', indicator=True)\n",
    "\n",
    "# Filter for rows that are only in the left or right dataframe\n",
    "uncommon_rows = merged[merged['_merge'] != 'both']\n",
    "\n",
    "# Drop the indicator column if not needed\n",
    "uncommon_rows = uncommon_rows.drop('_merge', axis=1)\n",
    "\n",
    "# Display the uncommon rows\n",
    "print(uncommon_rows)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = snp_data.merge(snp_data_old, how = 'inner' ,indicator=False) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos</th>\n",
       "      <th>chr_x</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad40_genome_AF_afr</th>\n",
       "      <th>gnomad40_genome_AF_ami</th>\n",
       "      <th>gnomad40_genome_AF_amr</th>\n",
       "      <th>gnomad40_genome_AF_asj</th>\n",
       "      <th>gnomad40_genome_AF_eas</th>\n",
       "      <th>gnomad40_genome_AF_fin</th>\n",
       "      <th>gnomad40_genome_AF_mid</th>\n",
       "      <th>gnomad40_genome_AF_nfe</th>\n",
       "      <th>gnomad40_genome_AF_remaining</th>\n",
       "      <th>gnomad40_genome_AF_sas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>32434</td>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.018746</td>\n",
       "      <td>165.452320</td>\n",
       "      <td>0.097936</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>47480</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015024</td>\n",
       "      <td>126.894470</td>\n",
       "      <td>0.070111</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>48195</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.019221</td>\n",
       "      <td>0.015570</td>\n",
       "      <td>77.986465</td>\n",
       "      <td>0.043584</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>...</td>\n",
       "      <td>2.407e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.411e-05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.0182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>48202</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.016036</td>\n",
       "      <td>115.160090</td>\n",
       "      <td>0.067440</td>\n",
       "      <td>3.284e-05</td>\n",
       "      <td>3.281e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>60400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>0.015248</td>\n",
       "      <td>79.551820</td>\n",
       "      <td>0.037764</td>\n",
       "      <td>1.314e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7033</th>\n",
       "      <td>197576155</td>\n",
       "      <td>chr3</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015796</td>\n",
       "      <td>94.429090</td>\n",
       "      <td>0.051905</td>\n",
       "      <td>2.74e-05</td>\n",
       "      <td>2.684e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>5.208e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.484e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7034</th>\n",
       "      <td>197636492</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015411</td>\n",
       "      <td>62.338760</td>\n",
       "      <td>0.026065</td>\n",
       "      <td>8.088e-05</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>...</td>\n",
       "      <td>5.003e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>6.805e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7035</th>\n",
       "      <td>197656793</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.055652</td>\n",
       "      <td>0.017351</td>\n",
       "      <td>84.323980</td>\n",
       "      <td>0.037448</td>\n",
       "      <td>0.1119</td>\n",
       "      <td>0.1683</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3134</td>\n",
       "      <td>0.1662</td>\n",
       "      <td>0.0692</td>\n",
       "      <td>0.0709</td>\n",
       "      <td>0.0022</td>\n",
       "      <td>0.0349</td>\n",
       "      <td>0.1206</td>\n",
       "      <td>0.0602</td>\n",
       "      <td>0.0888</td>\n",
       "      <td>0.0348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7036</th>\n",
       "      <td>198116021</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.241652</td>\n",
       "      <td>0.018788</td>\n",
       "      <td>87.795296</td>\n",
       "      <td>0.039123</td>\n",
       "      <td>0.2093</td>\n",
       "      <td>0.2097</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3173</td>\n",
       "      <td>0.0647</td>\n",
       "      <td>0.1589</td>\n",
       "      <td>0.1773</td>\n",
       "      <td>0.0702</td>\n",
       "      <td>0.2510</td>\n",
       "      <td>0.1939</td>\n",
       "      <td>0.1599</td>\n",
       "      <td>0.1744</td>\n",
       "      <td>0.2595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7037</th>\n",
       "      <td>198117279</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.016466</td>\n",
       "      <td>106.986080</td>\n",
       "      <td>0.056975</td>\n",
       "      <td>0.0301</td>\n",
       "      <td>0.1578</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0456</td>\n",
       "      <td>0.0062</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.0201</td>\n",
       "      <td>0.0087</td>\n",
       "      <td>0.0598</td>\n",
       "      <td>0.0287</td>\n",
       "      <td>0.0198</td>\n",
       "      <td>0.0278</td>\n",
       "      <td>0.0359</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7038 rows Ã— 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pos chr_x ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0         32434  chr3     A     C     0.000261         0.018746   \n",
       "1         47480  chr3     T     C     0.000174         0.015024   \n",
       "2         48195  chr3     T     C     0.019221         0.015570   \n",
       "3         48202  chr3     G     C     0.001739         0.016036   \n",
       "4         60400  chr3     G     C     0.001044         0.015248   \n",
       "...         ...   ...   ...   ...          ...              ...   \n",
       "7033  197576155  chr3     C     G     0.000174         0.015796   \n",
       "7034  197636492  chr3     T     C     0.000174         0.015411   \n",
       "7035  197656793  chr3     T     C     0.055652         0.017351   \n",
       "7036  198116021  chr3     T     C     0.241652         0.018788   \n",
       "7037  198117279  chr3     T     C     0.000177         0.016466   \n",
       "\n",
       "      log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                 165.452320        0.097936                  .   \n",
       "1                 126.894470        0.070111             0.0001   \n",
       "2                  77.986465        0.043584             0.0006   \n",
       "3                 115.160090        0.067440          3.284e-05   \n",
       "4                  79.551820        0.037764          1.314e-05   \n",
       "...                      ...             ...                ...   \n",
       "7033               94.429090        0.051905           2.74e-05   \n",
       "7034               62.338760        0.026065          8.088e-05   \n",
       "7035               84.323980        0.037448             0.1119   \n",
       "7036               87.795296        0.039123             0.2093   \n",
       "7037              106.986080        0.056975             0.0301   \n",
       "\n",
       "     gnomad40_genome_AF_raw  ... gnomad40_genome_AF_afr  \\\n",
       "0                         .  ...                      .   \n",
       "1                    0.0001  ...                 0.0004   \n",
       "2                    0.0006  ...              2.407e-05   \n",
       "3                 3.281e-05  ...                      0   \n",
       "4                 1.312e-05  ...                      0   \n",
       "...                     ...  ...                    ...   \n",
       "7033              2.684e-05  ...              5.208e-05   \n",
       "7034                 0.0011  ...              5.003e-05   \n",
       "7035                 0.1683  ...                 0.3134   \n",
       "7036                 0.2097  ...                 0.3173   \n",
       "7037                 0.1578  ...                 0.0456   \n",
       "\n",
       "     gnomad40_genome_AF_ami gnomad40_genome_AF_amr gnomad40_genome_AF_asj  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                 0.0001                      0   \n",
       "2                         0                 0.0001                      0   \n",
       "3                         0                      0                      0   \n",
       "4                         0                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "7033                      0                      0                      0   \n",
       "7034                      0              6.805e-05                      0   \n",
       "7035                 0.1662                 0.0692                 0.0709   \n",
       "7036                 0.0647                 0.1589                 0.1773   \n",
       "7037                 0.0062                 0.0308                 0.0201   \n",
       "\n",
       "     gnomad40_genome_AF_eas gnomad40_genome_AF_fin gnomad40_genome_AF_mid  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                      0                      0   \n",
       "2                         0                      0                      0   \n",
       "3                         0                      0                      0   \n",
       "4                    0.0002                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "7033                      0                      0                      0   \n",
       "7034                 0.0002                      0                      0   \n",
       "7035                 0.0022                 0.0349                 0.1206   \n",
       "7036                 0.0702                 0.2510                 0.1939   \n",
       "7037                 0.0087                 0.0598                 0.0287   \n",
       "\n",
       "     gnomad40_genome_AF_nfe gnomad40_genome_AF_remaining  \\\n",
       "0                         .                            .   \n",
       "1                         0                            0   \n",
       "2                 4.411e-05                       0.0009   \n",
       "3                  1.47e-05                            0   \n",
       "4                         0                            0   \n",
       "...                     ...                          ...   \n",
       "7033              1.484e-05                            0   \n",
       "7034                 0.0001                            0   \n",
       "7035                 0.0602                       0.0888   \n",
       "7036                 0.1599                       0.1744   \n",
       "7037                 0.0198                       0.0278   \n",
       "\n",
       "     gnomad40_genome_AF_sas  \n",
       "0                         .  \n",
       "1                         0  \n",
       "2                    0.0182  \n",
       "3                    0.0008  \n",
       "4                    0.0002  \n",
       "...                     ...  \n",
       "7033                 0.0002  \n",
       "7034                      0  \n",
       "7035                 0.0348  \n",
       "7036                 0.2595  \n",
       "7037                 0.0359  \n",
       "\n",
       "[7038 rows x 27 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_data.to_csv(os.path.join(directory3, 'gene_chr'+str(j)+'check.csv'),index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'DataFrame' object has no attribute 'duplicate'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-23-cceea367bcb8>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mtest\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msnp_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mduplicate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msubset\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"pos\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkeep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/apps/codes/anaconda3/lib/python3.7/site-packages/pandas-1.2.3-py3.7-linux-x86_64.egg/pandas/core/generic.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m   5463\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_info_axis\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_can_hold_identifiers_and_holds_name\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5464\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 5465\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   5466\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5467\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__setattr__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'DataFrame' object has no attribute 'duplicate'"
     ]
    }
   ],
   "source": [
    "test=snp_data.duplicate(subset=[\"pos\"], keep=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7037, 28)"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "test=snp_data[snp_data['pos'].duplicated(keep=False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "test1=data[data['pos'].duplicated(keep=False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>end</th>\n",
       "      <th>chr_anno</th>\n",
       "      <th>start_anno</th>\n",
       "      <th>end_anno</th>\n",
       "      <th>anno</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chr, pos, end, chr_anno, start_anno, end_anno, anno]\n",
       "Index: []"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "data[data['column_name'].duplicate(keep=False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_data=snp_data.drop_duplicates(subset=[\"pos\"], keep=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7037, 28)"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['pos', 'chr_x', 'ref_x', 'alt_x', 'allele_freq', 'log_counts_diff',\n",
       "       'log_probs_diff_abs_sum', 'probs_jsd_diff', 'gnomad40_genome_AF',\n",
       "       'gnomad40_genome_AF_raw', 'gnomad40_genome_AF_XX',\n",
       "       'gnomad40_genome_AF_XY', 'gnomad40_genome_AF_grpmax',\n",
       "       'gnomad40_genome_faf95', 'gnomad40_genome_faf99',\n",
       "       'gnomad40_genome_fafmax_faf95_max', 'gnomad40_genome_fafmax_faf99_max',\n",
       "       'gnomad40_genome_AF_afr', 'gnomad40_genome_AF_ami',\n",
       "       'gnomad40_genome_AF_amr', 'gnomad40_genome_AF_asj',\n",
       "       'gnomad40_genome_AF_eas', 'gnomad40_genome_AF_fin',\n",
       "       'gnomad40_genome_AF_mid', 'gnomad40_genome_AF_nfe',\n",
       "       'gnomad40_genome_AF_remaining', 'gnomad40_genome_AF_sas',\n",
       "       'CD4_Memory_Primary_Cells'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_data.to_csv(os.path.join(directory3, 'gene_chr'+str(j)+'.csv'),index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos</th>\n",
       "      <th>chr_x</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad40_genome_AF_ami</th>\n",
       "      <th>gnomad40_genome_AF_amr</th>\n",
       "      <th>gnomad40_genome_AF_asj</th>\n",
       "      <th>gnomad40_genome_AF_eas</th>\n",
       "      <th>gnomad40_genome_AF_fin</th>\n",
       "      <th>gnomad40_genome_AF_mid</th>\n",
       "      <th>gnomad40_genome_AF_nfe</th>\n",
       "      <th>gnomad40_genome_AF_remaining</th>\n",
       "      <th>gnomad40_genome_AF_sas</th>\n",
       "      <th>CD4_Memory_Primary_Cells</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>32434</td>\n",
       "      <td>chr3</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.018746</td>\n",
       "      <td>165.452320</td>\n",
       "      <td>0.097936</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>47480</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015024</td>\n",
       "      <td>126.894470</td>\n",
       "      <td>0.070111</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>48195</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.019221</td>\n",
       "      <td>0.015570</td>\n",
       "      <td>77.986465</td>\n",
       "      <td>0.043584</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.411e-05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.0182</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>48202</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.016036</td>\n",
       "      <td>115.160090</td>\n",
       "      <td>0.067440</td>\n",
       "      <td>3.284e-05</td>\n",
       "      <td>3.281e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0008</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>60400</td>\n",
       "      <td>chr3</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>0.015248</td>\n",
       "      <td>79.551820</td>\n",
       "      <td>0.037764</td>\n",
       "      <td>1.314e-05</td>\n",
       "      <td>1.312e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7032</th>\n",
       "      <td>197576155</td>\n",
       "      <td>chr3</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015796</td>\n",
       "      <td>94.429090</td>\n",
       "      <td>0.051905</td>\n",
       "      <td>2.74e-05</td>\n",
       "      <td>2.684e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.484e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>5_TxWk</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7033</th>\n",
       "      <td>197636492</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.015411</td>\n",
       "      <td>62.338760</td>\n",
       "      <td>0.026065</td>\n",
       "      <td>8.088e-05</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>6.805e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7034</th>\n",
       "      <td>197656793</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.055652</td>\n",
       "      <td>0.017351</td>\n",
       "      <td>84.323980</td>\n",
       "      <td>0.037448</td>\n",
       "      <td>0.1119</td>\n",
       "      <td>0.1683</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1662</td>\n",
       "      <td>0.0692</td>\n",
       "      <td>0.0709</td>\n",
       "      <td>0.0022</td>\n",
       "      <td>0.0349</td>\n",
       "      <td>0.1206</td>\n",
       "      <td>0.0602</td>\n",
       "      <td>0.0888</td>\n",
       "      <td>0.0348</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7035</th>\n",
       "      <td>198116021</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.241652</td>\n",
       "      <td>0.018788</td>\n",
       "      <td>87.795296</td>\n",
       "      <td>0.039123</td>\n",
       "      <td>0.2093</td>\n",
       "      <td>0.2097</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0647</td>\n",
       "      <td>0.1589</td>\n",
       "      <td>0.1773</td>\n",
       "      <td>0.0702</td>\n",
       "      <td>0.2510</td>\n",
       "      <td>0.1939</td>\n",
       "      <td>0.1599</td>\n",
       "      <td>0.1744</td>\n",
       "      <td>0.2595</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7036</th>\n",
       "      <td>198117279</td>\n",
       "      <td>chr3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.016466</td>\n",
       "      <td>106.986080</td>\n",
       "      <td>0.056975</td>\n",
       "      <td>0.0301</td>\n",
       "      <td>0.1578</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0062</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.0201</td>\n",
       "      <td>0.0087</td>\n",
       "      <td>0.0598</td>\n",
       "      <td>0.0287</td>\n",
       "      <td>0.0198</td>\n",
       "      <td>0.0278</td>\n",
       "      <td>0.0359</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7037 rows Ã— 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pos chr_x ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0         32434  chr3     A     C     0.000261         0.018746   \n",
       "1         47480  chr3     T     C     0.000174         0.015024   \n",
       "2         48195  chr3     T     C     0.019221         0.015570   \n",
       "3         48202  chr3     G     C     0.001739         0.016036   \n",
       "4         60400  chr3     G     C     0.001044         0.015248   \n",
       "...         ...   ...   ...   ...          ...              ...   \n",
       "7032  197576155  chr3     C     G     0.000174         0.015796   \n",
       "7033  197636492  chr3     T     C     0.000174         0.015411   \n",
       "7034  197656793  chr3     T     C     0.055652         0.017351   \n",
       "7035  198116021  chr3     T     C     0.241652         0.018788   \n",
       "7036  198117279  chr3     T     C     0.000177         0.016466   \n",
       "\n",
       "      log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                 165.452320        0.097936                  .   \n",
       "1                 126.894470        0.070111             0.0001   \n",
       "2                  77.986465        0.043584             0.0006   \n",
       "3                 115.160090        0.067440          3.284e-05   \n",
       "4                  79.551820        0.037764          1.314e-05   \n",
       "...                      ...             ...                ...   \n",
       "7032               94.429090        0.051905           2.74e-05   \n",
       "7033               62.338760        0.026065          8.088e-05   \n",
       "7034               84.323980        0.037448             0.1119   \n",
       "7035               87.795296        0.039123             0.2093   \n",
       "7036              106.986080        0.056975             0.0301   \n",
       "\n",
       "     gnomad40_genome_AF_raw  ... gnomad40_genome_AF_ami  \\\n",
       "0                         .  ...                      .   \n",
       "1                    0.0001  ...                      0   \n",
       "2                    0.0006  ...                      0   \n",
       "3                 3.281e-05  ...                      0   \n",
       "4                 1.312e-05  ...                      0   \n",
       "...                     ...  ...                    ...   \n",
       "7032              2.684e-05  ...                      0   \n",
       "7033                 0.0011  ...                      0   \n",
       "7034                 0.1683  ...                 0.1662   \n",
       "7035                 0.2097  ...                 0.0647   \n",
       "7036                 0.1578  ...                 0.0062   \n",
       "\n",
       "     gnomad40_genome_AF_amr gnomad40_genome_AF_asj gnomad40_genome_AF_eas  \\\n",
       "0                         .                      .                      .   \n",
       "1                    0.0001                      0                      0   \n",
       "2                    0.0001                      0                      0   \n",
       "3                         0                      0                      0   \n",
       "4                         0                      0                 0.0002   \n",
       "...                     ...                    ...                    ...   \n",
       "7032                      0                      0                      0   \n",
       "7033              6.805e-05                      0                 0.0002   \n",
       "7034                 0.0692                 0.0709                 0.0022   \n",
       "7035                 0.1589                 0.1773                 0.0702   \n",
       "7036                 0.0308                 0.0201                 0.0087   \n",
       "\n",
       "     gnomad40_genome_AF_fin gnomad40_genome_AF_mid gnomad40_genome_AF_nfe  \\\n",
       "0                         .                      .                      .   \n",
       "1                         0                      0                      0   \n",
       "2                         0                      0              4.411e-05   \n",
       "3                         0                      0               1.47e-05   \n",
       "4                         0                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "7032                      0                      0              1.484e-05   \n",
       "7033                      0                      0                 0.0001   \n",
       "7034                 0.0349                 0.1206                 0.0602   \n",
       "7035                 0.2510                 0.1939                 0.1599   \n",
       "7036                 0.0598                 0.0287                 0.0198   \n",
       "\n",
       "     gnomad40_genome_AF_remaining gnomad40_genome_AF_sas  \\\n",
       "0                               .                      .   \n",
       "1                               0                      0   \n",
       "2                          0.0009                 0.0182   \n",
       "3                               0                 0.0008   \n",
       "4                               0                 0.0002   \n",
       "...                           ...                    ...   \n",
       "7032                            0                 0.0002   \n",
       "7033                            0                      0   \n",
       "7034                       0.0888                 0.0348   \n",
       "7035                       0.1744                 0.2595   \n",
       "7036                       0.0278                 0.0359   \n",
       "\n",
       "     CD4_Memory_Primary_Cells  \n",
       "0                    15_Quies  \n",
       "1                    15_Quies  \n",
       "2                    15_Quies  \n",
       "3                    15_Quies  \n",
       "4                    15_Quies  \n",
       "...                       ...  \n",
       "7032                   5_TxWk  \n",
       "7033                 15_Quies  \n",
       "7034                 15_Quies  \n",
       "7035                      NaN  \n",
       "7036                      NaN  \n",
       "\n",
       "[7037 rows x 28 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene_chr3_E025_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E128_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E119_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E030_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E029_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E059_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E077_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E091_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E107_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E039_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E097_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E016_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E096_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E048_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E072_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E019_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E024_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E073_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E052_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E051_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E063_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E083_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E090_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E058_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E088_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E041_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E099_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E010_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E047_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E123_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E105_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E018_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E098_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E044_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E038_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E068_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E081_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E067_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E108_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E020_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E127_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E021_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E093_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E095_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E031_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E012_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E036_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E115_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E114_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E023_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E112_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E055_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E071_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E092_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E087_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E121_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E045_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E085_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E122_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E106_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E078_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E028_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E046_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E034_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E126_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E057_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E040_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E049_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E080_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E033_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E054_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E001_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E116_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E069_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E101_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E109_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E002_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E079_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E103_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E013_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E102_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E015_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E074_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E022_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E066_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E027_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E118_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E062_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E037_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E075_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E125_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E043_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E100_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E113_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E056_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E111_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E014_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E124_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E065_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E104_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E053_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E117_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E070_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E061_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E076_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E082_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E086_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E120_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E084_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E003_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E094_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E110_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E011_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E035_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E032_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E050_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E017_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E129_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E089_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E026_15_coreMarks_dense.bed.bgz.vcf\n",
      "gene_chr3_E042_15_coreMarks_dense.bed.bgz.vcf\n",
      "1\n",
      "2\n",
      "3\n"
     ]
    },
    {
     "ename": "MemoryError",
     "evalue": "Unable to allocate 92.9 GiB for an array with shape (12465660166,) and data type int64",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mMemoryError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-76-943a7b1a474b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     30\u001b[0m     \u001b[0mnam\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mre\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msearch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"_E\\d*_\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfiles\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstrip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"_\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     31\u001b[0m     \u001b[0mtiss_nam\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mheaders\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mheaders\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mid\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mnam\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'tissue'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 32\u001b[0;31m     \u001b[0msnp_data\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmerge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msnp_data\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mall_data\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"pos\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"anno\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mon\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'pos'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'left'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     33\u001b[0m     \u001b[0msnp_data\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnp_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m\"anno\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mtiss_nam\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     34\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/apps/codes/anaconda3/lib/python3.7/site-packages/pandas-1.2.3-py3.7-linux-x86_64.egg/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36mmerge\u001b[0;34m(left, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)\u001b[0m\n\u001b[1;32m     87\u001b[0m         \u001b[0mvalidate\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mvalidate\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     88\u001b[0m     )\n\u001b[0;32m---> 89\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     90\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     91\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/apps/codes/anaconda3/lib/python3.7/site-packages/pandas-1.2.3-py3.7-linux-x86_64.egg/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36mget_result\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    682\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mleft\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mright\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_indicator_pre_merge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mleft\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mright\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    683\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 684\u001b[0;31m         \u001b[0mjoin_index\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mleft_indexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mright_indexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_join_info\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    685\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    686\u001b[0m         llabels, rlabels = _items_overlap_with_suffix(\n",
      "\u001b[0;32m/apps/codes/anaconda3/lib/python3.7/site-packages/pandas-1.2.3-py3.7-linux-x86_64.egg/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36m_get_join_info\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    907\u001b[0m             )\n\u001b[1;32m    908\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 909\u001b[0;31m             \u001b[0;34m(\u001b[0m\u001b[0mleft_indexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mright_indexer\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_join_indexers\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    910\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    911\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mright_index\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/apps/codes/anaconda3/lib/python3.7/site-packages/pandas-1.2.3-py3.7-linux-x86_64.egg/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36m_get_join_indexers\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    886\u001b[0m         \u001b[0;34m\"\"\" return the join indexers \"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    887\u001b[0m         return get_join_indexers(\n\u001b[0;32m--> 888\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mleft_join_keys\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mright_join_keys\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msort\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msort\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhow\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    889\u001b[0m         )\n\u001b[1;32m    890\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/apps/codes/anaconda3/lib/python3.7/site-packages/pandas-1.2.3-py3.7-linux-x86_64.egg/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36mget_join_indexers\u001b[0;34m(left_keys, right_keys, sort, how, **kwargs)\u001b[0m\n\u001b[1;32m   1439\u001b[0m     }[how]\n\u001b[1;32m   1440\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1441\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mjoin_func\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcount\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1442\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1443\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/join.pyx\u001b[0m in \u001b[0;36mpandas._libs.join.left_outer_join\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mMemoryError\u001b[0m: Unable to allocate 92.9 GiB for an array with shape (12465660166,) and data type int64"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "directory3='/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model1/final_data/'\n",
    " # Replace with the actual output directory\n",
    "\n",
    "files = os.listdir('model1/anno_af_chrom3')\n",
    "headers = pd.read_csv(\"sampleids_new_sorted.txt\", ' ', names=['id', 'tissue'])\n",
    "\n",
    "# Read snp_data file outside the loop\n",
    "j = 3  # or any other value\n",
    "file_snp_data = f\"allele_freq_chr{j}_gene_ncr.tsv_chr_{j}.csv\"\n",
    "snp_data = pd.read_csv(f\"/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model1/anno_af_chrom_new/{file_snp_data}\")\n",
    "\n",
    "# Create a list to store data frames\n",
    "data_frames = []\n",
    "\n",
    "for i, file_name in enumerate(files):\n",
    "    if file_name.startswith(f'gene_chr{j}_'):\n",
    "        print(file_name)\n",
    "        data = pd.read_csv(f\"model1/anno_af_chrom3/{file_name}\", '\\t',\n",
    "                           names=['chr', 'pos', 'end', 'chr_anno', 'start_anno', 'end_anno', 'anno'])\n",
    "        data_frames.append(data)\n",
    "\n",
    "# Concatenate all data frames into a single data frame\n",
    "all_data = pd.concat(data_frames, ignore_index=True)\n",
    "\n",
    "for i in range(1, 121):\n",
    "    print(i)\n",
    "    nam = re.search(\"_E\\d*_\", files[i])[0].strip(\"_\")\n",
    "    tiss_nam = headers[headers.id == nam]['tissue'].values[0]\n",
    "    snp_data = pd.merge(snp_data, all_data[[\"pos\", \"anno\"]], on=['pos'], how='left')\n",
    "    snp_data = snp_data.rename(columns={\"anno\": tiss_nam})\n",
    "\n",
    "snp_data = snp_data.drop_duplicates(subset=[\"pos\"], keep=False)\n",
    "snp_data.to_csv(os.path.join(directory3, f'gene_chr{j}.csv'), index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "2\n",
      "3\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-19-c3559242490d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     31\u001b[0m         \u001b[0;31m#tiss_nam = headers[headers.id == nam]['tissue'].values[0]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     32\u001b[0m         \u001b[0;31m#print(nam,tiss_nam)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 33\u001b[0;31m         \u001b[0msnp_data\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmerge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msnp_data\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"pos\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"anno\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mon\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'pos'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'left'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     34\u001b[0m         \u001b[0msnp_data\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnp_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m\"anno\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mtiss_nam\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     35\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/apps/codes/anaconda3/lib/python3.7/site-packages/pandas-1.2.3-py3.7-linux-x86_64.egg/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36mmerge\u001b[0;34m(left, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)\u001b[0m\n\u001b[1;32m     87\u001b[0m         \u001b[0mvalidate\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mvalidate\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     88\u001b[0m     )\n\u001b[0;32m---> 89\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     90\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     91\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/apps/codes/anaconda3/lib/python3.7/site-packages/pandas-1.2.3-py3.7-linux-x86_64.egg/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36mget_result\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    695\u001b[0m             \u001b[0maxes\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mllabels\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrlabels\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mjoin_index\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    696\u001b[0m             \u001b[0mconcat_axis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 697\u001b[0;31m             \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    698\u001b[0m         )\n\u001b[1;32m    699\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/apps/codes/anaconda3/lib/python3.7/site-packages/pandas-1.2.3-py3.7-linux-x86_64.egg/pandas/core/internals/concat.py\u001b[0m in \u001b[0;36mconcatenate_block_managers\u001b[0;34m(mgrs_indexers, axes, concat_axis, copy)\u001b[0m\n\u001b[1;32m     81\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     82\u001b[0m             b = make_block(\n\u001b[0;32m---> 83\u001b[0;31m                 \u001b[0m_concatenate_join_units\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mjoin_units\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconcat_axis\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     84\u001b[0m                 \u001b[0mplacement\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mplacement\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     85\u001b[0m                 \u001b[0mndim\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0maxes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/apps/codes/anaconda3/lib/python3.7/site-packages/pandas-1.2.3-py3.7-linux-x86_64.egg/pandas/core/internals/concat.py\u001b[0m in \u001b[0;36m_concatenate_join_units\u001b[0;34m(join_units, concat_axis, copy)\u001b[0m\n\u001b[1;32m    313\u001b[0m     to_concat = [\n\u001b[1;32m    314\u001b[0m         \u001b[0mju\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_reindexed_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mempty_dtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mempty_dtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mupcasted_na\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mupcasted_na\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 315\u001b[0;31m         \u001b[0;32mfor\u001b[0m \u001b[0mju\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mjoin_units\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    316\u001b[0m     ]\n\u001b[1;32m    317\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/apps/codes/anaconda3/lib/python3.7/site-packages/pandas-1.2.3-py3.7-linux-x86_64.egg/pandas/core/internals/concat.py\u001b[0m in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m    313\u001b[0m     to_concat = [\n\u001b[1;32m    314\u001b[0m         \u001b[0mju\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_reindexed_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mempty_dtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mempty_dtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mupcasted_na\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mupcasted_na\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 315\u001b[0;31m         \u001b[0;32mfor\u001b[0m \u001b[0mju\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mjoin_units\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    316\u001b[0m     ]\n\u001b[1;32m    317\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/apps/codes/anaconda3/lib/python3.7/site-packages/pandas-1.2.3-py3.7-linux-x86_64.egg/pandas/core/internals/concat.py\u001b[0m in \u001b[0;36mget_reindexed_values\u001b[0;34m(self, empty_dtype, upcasted_na)\u001b[0m\n\u001b[1;32m    296\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    297\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0max\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindexer\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindexers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 298\u001b[0;31m                 \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0malgos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtake_nd\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0max\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfill_value\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfill_value\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    299\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    300\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mvalues\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/apps/codes/anaconda3/lib/python3.7/site-packages/pandas-1.2.3-py3.7-linux-x86_64.egg/pandas/core/algorithms.py\u001b[0m in \u001b[0;36mtake_nd\u001b[0;34m(arr, indexer, axis, out, fill_value, allow_fill)\u001b[0m\n\u001b[1;32m   1752\u001b[0m             \u001b[0mout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mempty\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mout_shape\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morder\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"F\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1753\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1754\u001b[0;31m             \u001b[0mout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mempty\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mout_shape\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1755\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1756\u001b[0m     func = _get_take_nd_function(\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "directory3 = '/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model1/final_data/'  # Replace with the actual output directory\n",
    "\n",
    "files = os.listdir('model1/anno_af_chrom3')\n",
    "headers = pd.read_csv(\"sampleids_new_sorted.txt\", ' ', names=['id', 'tissue'])\n",
    "\n",
    "# Read the snp_data file outside the loop if it's the same for all j\n",
    "#snp_data = \"/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model1/anno_af_chrom_new/allele_freq_chr{j}_gene_ncr.tsv_chr_{j}.csv\"\n",
    "#snp_data = pd.read_csv(file_snp_data.format(1))  # Assuming it's the same for all j\n",
    "\n",
    "for j in range(1, 23):\n",
    "    \n",
    "    files2 = [x for x in files if x.startswith(f'gene_chr{j}_')]\n",
    "    print(j)\n",
    "    #print(len(files2))\n",
    "    file=f'allele_freq_chr{j}_gene_ncr.tsv_chr_{j}.csv'\n",
    "    snp_data=pd.read_csv(\"/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model1/anno_af_chrom_new/\"+file)\n",
    "    \n",
    "    for i, file_name in enumerate(files2):\n",
    "        #print(file_name)\n",
    "        data = pd.read_csv(\"model1/anno_af_chrom3/\" + file_name, '\\t',\n",
    "                           names=['chr', 'pos', 'end', 'chr_anno', 'start_anno', 'end_anno', 'anno'])\n",
    "        \n",
    "        nam=re.search(\"_E\\d*_\",files2[i])[0]\n",
    "        nam=nam.strip(\"_\")\n",
    "        tiss_nam=headers[headers.id==nam]['tissue'].values[0]\n",
    "        \n",
    "        #nam = re.search(\"_E\\d*_\", file_name)[0].strip(\"_\")\n",
    "        #tiss_nam = headers[headers.id == nam]['tissue'].values[0]\n",
    "        #print(nam,tiss_nam)\n",
    "        snp_data = pd.merge(snp_data, data[[\"pos\", \"anno\"]], on=['pos'], how='left')\n",
    "        snp_data = snp_data.rename(columns={\"anno\": tiss_nam})\n",
    "\n",
    "    snp_data = snp_data.drop_duplicates(subset=[\"pos\"], keep=False)\n",
    "    snp_data.to_csv(os.path.join(directory3, f'gene_chr{j}.csv'), index=None)\n",
    "    del snp_data\n",
    "    del files2\n",
    "    del data\n",
    "    gc.collect()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#tis_data=pd.read_csv(\"/gpfs/data/user/shweta_lab/data/roadmap/egg2.wustl.edu/roadmap/withgh38liftover/E001_15_coreMarks_dense.bed.bgz\",sep='\\t',compression=\"gzip\",usecols=[0,1,2,3],names=['chr','start','end','Anno'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#headers=pd.read_csv(\"sampleids_new_sorted.txt\",' ',names=['id','tissue'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {},
   "outputs": [],
   "source": [
    "# filename_af1=[]\n",
    "# for filename_af in os.listdir(directory2):\n",
    "#     if '_gene_' in filename_af:\n",
    "#     #if filename_af.with('_gene_'):\n",
    "#         with open(os.path.join(directory2, filename_af)) as f:\n",
    "#             filename_af1.append(filename_af)\n",
    "# print(filename_af1)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# for i in range(1,23):\n",
    "#     file=f'allele_freq_chr{i}_gene_ncr.tsv_chr_{i}.csv'\n",
    "#     print(file)\n",
    "#     break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "# def extract_integer(file_name):\n",
    "#     return int(re.search(r'\\d+', file_name).group())\n",
    "\n",
    "# # Sort the file names based on the extracted integer\n",
    "# sorted_filename_af1= sorted(filename_af1, key=extract_integer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sorted_filename_af1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "# files2=[]\n",
    "# for x in files:\n",
    "#     if x.startswith('output_chr1_'):\n",
    "#         files2.append(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'output_chr1_E092_15_coreMarks_dense.bed.bgz.vcf'"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# files2[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Define a sorting key function to extract the integer from the file name\n",
    "# def extract_integer(file_name):\n",
    "#     return int(re.search(r'\\d+', file_name).group())\n",
    "\n",
    "# # Sort the file names based on the extracted integer\n",
    "# sorted_files= sorted(files2, key=extract_integer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "#sorted_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "# data=[]\n",
    "# for l in sorted_filename_af1:\n",
    "#     #print(l[12:16])\n",
    "#     for i,k in zip(headers['id'],headers['tissue']):\n",
    "#         for j in sorted_files:\n",
    "#             #print(j[7:11])\n",
    "#             if (i==j[12:16] and j[7:12]==l[12:17]):\n",
    "                \n",
    "#                 full_path = os.path.join(directory, j)\n",
    "                \n",
    "#                 full_path1 = os.path.join(directory2, l)\n",
    "                \n",
    "#                 old_data=pd.read_csv(full_path1,',')\n",
    "                \n",
    "#                 #snp_data=pd.read_csv(full_path,'\\t',names=['chr','pos','end','chr_anno','start_anno','end_anno','anno'])\n",
    "#                 #snp_data=snp_data[['chr_x','start','end_x','anno_x','anno_y']]\n",
    "#                 #data=pd.merge(snp_data, data, on=['start'], how='outer',)\n",
    "\n",
    "#                 #snp_data=pd.merge(snp_data, data, on=['start'], how='outer',)\n",
    "\n",
    "\n",
    "#                 print(old_data)\n",
    "\n",
    "#                 #print(i,k,j)\n",
    "#                 #print(i,j)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [],
   "source": [
    "# snp_data=pd.read_csv(\"/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model1/anno_af_chrom/\"+file)\n",
    "# headers=pd.read_csv(\"sampleids_new_sorted.txt\",' ',names=['id','tissue'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [],
   "source": [
    "# data=pd.read_csv(\"model1/anno_af_chrom3/\"+files2[0],'\\t',names=['chr','pos','end','chr_anno','start_anno','end_anno','anno'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [],
   "source": [
    "# nam=re.search(\"_E\\d*_\",files2[0])[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [],
   "source": [
    "# nam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [],
   "source": [
    "# nam=nam.strip(\"_\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "# tiss_nam=headers[headers.id==nam]['tissue'].values[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "# snp_data=pd.merge(snp_data, data[['pos','anno']], on=['pos'], how='left',)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {},
   "outputs": [],
   "source": [
    "# snp_data.rename(columns={\"anno\":tiss_nam})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# files2=[]\n",
    "\n",
    "# for x in files:\n",
    "    \n",
    "#     if x.startswith('output_chr1_'):\n",
    "        \n",
    "#         files2.append(x)\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'output_chr1_E019_15_coreMarks_dense.bed.bgz.vcf'"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files2[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "metadata": {},
   "outputs": [],
   "source": [
    "headers=pd.read_csv(\"sampleids_new_sorted.txt\",' ',names=['id','tissue'])\n",
    "for j in range(1,2):\n",
    "    files2=[]\n",
    "    for x in files:\n",
    "        if x.startswith(f'gene_chr{j}_'):\n",
    "            files2.append(x)\n",
    "            \n",
    "    file=f'allele_freq_chr{j}_gene_ncr.tsv_chr_{j}.csv'\n",
    "    snp_data=pd.read_csv(\"/gpfs/data/user/himanshu/chrombpnet_old/bash_scripts/snp_pred_191223/analysis/model1/anno_af_chrom_new/\"+file)\n",
    "    \n",
    "    for i in range(121):\n",
    "        data=pd.read_csv(\"model1/anno_af_chrom3/\"+files2[i],'\\t',names=['chr','pos','end','chr_anno','start_anno','end_anno','anno'])\n",
    "        nam=re.search(\"_E\\d*_\",files2[i])[0]\n",
    "        nam=nam.strip(\"_\")\n",
    "        tiss_nam=headers[headers.id==nam]['tissue'].values[0]\n",
    "        #print(tiss_nam)\n",
    "\n",
    "        snp_data=pd.merge(snp_data, data[[\"pos\",\"anno\"]], on=['pos'], how='left')\n",
    "        snp_data=snp_data.rename(columns={\"anno\":tiss_nam})\n",
    "    snp_data=snp_data.drop_duplicates(subset=[\"pos\"], keep=False)\n",
    "    snp_data.to_csv(os.path.join(directory3, 'gene_chr'+str(i)+'.csv'),index=None)\n",
    "        #print(l)\n",
    "        #print(l,files2[i])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['output_chr1_E092_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E019_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E113_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E083_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E041_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E128_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E091_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E071_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E015_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E116_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E039_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E073_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E118_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E035_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E020_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E119_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E040_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E125_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E087_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E101_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E030_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E048_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E037_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E022_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E036_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E126_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E089_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E074_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E097_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E085_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E111_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E038_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E086_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E034_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E094_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E029_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E099_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E056_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E072_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E061_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E095_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E124_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E050_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E053_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E026_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E046_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E076_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E032_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E063_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E011_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E021_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E103_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E052_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E010_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E075_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E078_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E002_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E024_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E044_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E012_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E077_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E033_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E001_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E100_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E070_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E117_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E096_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E079_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E049_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E102_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E031_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E062_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E045_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E098_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E003_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E069_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E115_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E123_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E027_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E016_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E047_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E110_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E054_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E104_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E121_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E018_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E051_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E112_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E109_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E013_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E023_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E058_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E120_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E025_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E028_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E059_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E043_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E055_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E122_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E065_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E066_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E057_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E084_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E067_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E068_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E080_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E127_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E014_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E088_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E093_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E107_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E082_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E129_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E017_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E106_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E114_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E090_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E105_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E081_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E108_15_coreMarks_dense.bed.bgz.vcf',\n",
       " 'output_chr1_E042_15_coreMarks_dense.bed.bgz.vcf']"
      ]
     },
     "execution_count": 251,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr_x</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>Fetal_Brain_Female</th>\n",
       "      <th>Osteoblasts</th>\n",
       "      <th>IMR90_Cell_Line</th>\n",
       "      <th>Sigmoid_Colon</th>\n",
       "      <th>A549_EtOH_0.02pct_Lung_Carcinoma</th>\n",
       "      <th>Fetal_Muscle_Leg</th>\n",
       "      <th>Right_Ventricle</th>\n",
       "      <th>Fetal_Brain_Male</th>\n",
       "      <th>Skeletal_Muscle_Female</th>\n",
       "      <th>CD4+_CD25-_IL17+_PMA-Ionomcyin_stimulated_Th17_Primary_Cells</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>77502</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000761</td>\n",
       "      <td>0.014884</td>\n",
       "      <td>66.712830</td>\n",
       "      <td>0.027054</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>...</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>86065</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.069599</td>\n",
       "      <td>0.014811</td>\n",
       "      <td>189.841190</td>\n",
       "      <td>0.106602</td>\n",
       "      <td>0.0869</td>\n",
       "      <td>0.0892</td>\n",
       "      <td>...</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>14_ReprPCWk</td>\n",
       "      <td>14_ReprPCWk</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>14_ReprPCWk</td>\n",
       "      <td>14_ReprPCWk</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>883806</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001913</td>\n",
       "      <td>0.015814</td>\n",
       "      <td>86.104660</td>\n",
       "      <td>0.048685</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>...</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>894725</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.016190</td>\n",
       "      <td>104.184900</td>\n",
       "      <td>0.052455</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>...</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>5_TxWk</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>7_Enh</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>1_TssA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>988951</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.014875</td>\n",
       "      <td>71.228340</td>\n",
       "      <td>0.038546</td>\n",
       "      <td>2.67e-05</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>...</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>5_TxWk</td>\n",
       "      <td>14_ReprPCWk</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15318</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248877982</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002087</td>\n",
       "      <td>0.020471</td>\n",
       "      <td>109.172440</td>\n",
       "      <td>0.046729</td>\n",
       "      <td>0.0035</td>\n",
       "      <td>0.0035</td>\n",
       "      <td>...</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15319</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248879155</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002696</td>\n",
       "      <td>0.015206</td>\n",
       "      <td>61.615257</td>\n",
       "      <td>0.028550</td>\n",
       "      <td>0.0285</td>\n",
       "      <td>0.0285</td>\n",
       "      <td>...</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15320</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248885752</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000696</td>\n",
       "      <td>0.018184</td>\n",
       "      <td>145.333740</td>\n",
       "      <td>0.086757</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>...</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15321</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248888164</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.002261</td>\n",
       "      <td>0.015049</td>\n",
       "      <td>97.845090</td>\n",
       "      <td>0.056807</td>\n",
       "      <td>0.0091</td>\n",
       "      <td>0.0091</td>\n",
       "      <td>...</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15322</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248893146</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.015502</td>\n",
       "      <td>78.590880</td>\n",
       "      <td>0.031837</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6811 rows Ã— 148 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr_x        pos ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0      chr1      77502     A     G     0.000761         0.014884   \n",
       "1      chr1      86065     G     C     0.069599         0.014811   \n",
       "2      chr1     883806     T     C     0.001913         0.015814   \n",
       "3      chr1     894725     T     C     0.000261         0.016190   \n",
       "4      chr1     988951     T     C     0.000174         0.014875   \n",
       "...     ...        ...   ...   ...          ...              ...   \n",
       "15318  chr1  248877982     A     C     0.002087         0.020471   \n",
       "15319  chr1  248879155     T     C     0.002696         0.015206   \n",
       "15320  chr1  248885752     T     C     0.000696         0.018184   \n",
       "15321  chr1  248888164     A     G     0.002261         0.015049   \n",
       "15322  chr1  248893146     A     C     0.000261         0.015502   \n",
       "\n",
       "       log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                   66.712830        0.027054             0.0002   \n",
       "1                  189.841190        0.106602             0.0869   \n",
       "2                   86.104660        0.048685             0.0006   \n",
       "3                  104.184900        0.052455             0.0001   \n",
       "4                   71.228340        0.038546           2.67e-05   \n",
       "...                       ...             ...                ...   \n",
       "15318              109.172440        0.046729             0.0035   \n",
       "15319               61.615257        0.028550             0.0285   \n",
       "15320              145.333740        0.086757             0.0002   \n",
       "15321               97.845090        0.056807             0.0091   \n",
       "15322               78.590880        0.031837                  .   \n",
       "\n",
       "      gnomad40_genome_AF_raw  ... Fetal_Brain_Female  Osteoblasts  \\\n",
       "0                     0.0002  ...           15_Quies     15_Quies   \n",
       "1                     0.0892  ...           15_Quies  14_ReprPCWk   \n",
       "2                     0.0007  ...               4_Tx         4_Tx   \n",
       "3                     0.0001  ...             1_TssA       1_TssA   \n",
       "4                     0.0004  ...               4_Tx     15_Quies   \n",
       "...                      ...  ...                ...          ...   \n",
       "15318                 0.0035  ...              9_Het     15_Quies   \n",
       "15319                 0.0285  ...              9_Het     15_Quies   \n",
       "15320                 0.0002  ...              9_Het     15_Quies   \n",
       "15321                 0.0091  ...           15_Quies     15_Quies   \n",
       "15322                      .  ...              9_Het     15_Quies   \n",
       "\n",
       "      IMR90_Cell_Line Sigmoid_Colon A549_EtOH_0.02pct_Lung_Carcinoma  \\\n",
       "0            15_Quies      15_Quies                         15_Quies   \n",
       "1         14_ReprPCWk      15_Quies                         15_Quies   \n",
       "2                4_Tx          4_Tx                             4_Tx   \n",
       "3              1_TssA        5_TxWk                           1_TssA   \n",
       "4                4_Tx          4_Tx                             4_Tx   \n",
       "...               ...           ...                              ...   \n",
       "15318        15_Quies         9_Het                         15_Quies   \n",
       "15319        15_Quies      15_Quies                         15_Quies   \n",
       "15320        15_Quies      15_Quies                         15_Quies   \n",
       "15321        15_Quies      15_Quies                         15_Quies   \n",
       "15322           9_Het      15_Quies                         15_Quies   \n",
       "\n",
       "      Fetal_Muscle_Leg Right_Ventricle Fetal_Brain_Male  \\\n",
       "0             15_Quies        15_Quies         15_Quies   \n",
       "1             15_Quies        15_Quies      14_ReprPCWk   \n",
       "2                 4_Tx            4_Tx         15_Quies   \n",
       "3               1_TssA          1_TssA            7_Enh   \n",
       "4                 4_Tx            4_Tx           5_TxWk   \n",
       "...                ...             ...              ...   \n",
       "15318            9_Het        15_Quies            9_Het   \n",
       "15319            9_Het        15_Quies         15_Quies   \n",
       "15320            9_Het        15_Quies         15_Quies   \n",
       "15321         15_Quies        15_Quies         15_Quies   \n",
       "15322            9_Het        15_Quies         15_Quies   \n",
       "\n",
       "      Skeletal_Muscle_Female  \\\n",
       "0                   15_Quies   \n",
       "1                14_ReprPCWk   \n",
       "2                       4_Tx   \n",
       "3                     1_TssA   \n",
       "4                14_ReprPCWk   \n",
       "...                      ...   \n",
       "15318               15_Quies   \n",
       "15319               15_Quies   \n",
       "15320               15_Quies   \n",
       "15321               15_Quies   \n",
       "15322               15_Quies   \n",
       "\n",
       "      CD4+_CD25-_IL17+_PMA-Ionomcyin_stimulated_Th17_Primary_Cells  \n",
       "0                                               15_Quies            \n",
       "1                                               15_Quies            \n",
       "2                                                   4_Tx            \n",
       "3                                                 1_TssA            \n",
       "4                                               15_Quies            \n",
       "...                                                  ...            \n",
       "15318                                           15_Quies            \n",
       "15319                                           15_Quies            \n",
       "15320                                           15_Quies            \n",
       "15321                                           15_Quies            \n",
       "15322                                           15_Quies            \n",
       "\n",
       "[6811 rows x 148 columns]"
      ]
     },
     "execution_count": 223,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data.drop_duplicates(subset=[\"pos\"], keep=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr_x</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>Duodenum_Mucosa</th>\n",
       "      <th>CD3_Primary_Cells_Cord_BI</th>\n",
       "      <th>ES-I3_Cell_Line</th>\n",
       "      <th>Psoas_Muscle</th>\n",
       "      <th>Brain_Germinal_Matrix</th>\n",
       "      <th>HeLa-S3_Cervical_Carcinoma</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Esophagus</th>\n",
       "      <th>Chondrocytes_from_Bone_Marrow_Derived_Mesenchymal_Stem_Cell_Cultured_Cells</th>\n",
       "      <th>Rectal_Mucosa.Donor_31</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>77502</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000761</td>\n",
       "      <td>0.014884</td>\n",
       "      <td>66.712830</td>\n",
       "      <td>0.027054</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>...</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>86065</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.069599</td>\n",
       "      <td>0.014811</td>\n",
       "      <td>189.841190</td>\n",
       "      <td>0.106602</td>\n",
       "      <td>0.0869</td>\n",
       "      <td>0.0892</td>\n",
       "      <td>...</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>14_ReprPCWk</td>\n",
       "      <td>14_ReprPCWk</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>14_ReprPCWk</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>883806</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001913</td>\n",
       "      <td>0.015814</td>\n",
       "      <td>86.104660</td>\n",
       "      <td>0.048685</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>...</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>5_TxWk</td>\n",
       "      <td>5_TxWk</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>5_TxWk</td>\n",
       "      <td>8_ZNF/Rpts</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>894725</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.016190</td>\n",
       "      <td>104.184900</td>\n",
       "      <td>0.052455</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>...</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>13_ReprPC</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>10_TssBiv</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>1_TssA</td>\n",
       "      <td>1_TssA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>988951</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.014875</td>\n",
       "      <td>71.228340</td>\n",
       "      <td>0.038546</td>\n",
       "      <td>2.67e-05</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>...</td>\n",
       "      <td>5_TxWk</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>5_TxWk</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>5_TxWk</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "      <td>4_Tx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7668</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248877982</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002087</td>\n",
       "      <td>0.020471</td>\n",
       "      <td>109.172440</td>\n",
       "      <td>0.046729</td>\n",
       "      <td>0.0035</td>\n",
       "      <td>0.0035</td>\n",
       "      <td>...</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7669</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248879155</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002696</td>\n",
       "      <td>0.015206</td>\n",
       "      <td>61.615257</td>\n",
       "      <td>0.028550</td>\n",
       "      <td>0.0285</td>\n",
       "      <td>0.0285</td>\n",
       "      <td>...</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7670</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248885752</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000696</td>\n",
       "      <td>0.018184</td>\n",
       "      <td>145.333740</td>\n",
       "      <td>0.086757</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>...</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7671</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248888164</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.002261</td>\n",
       "      <td>0.015049</td>\n",
       "      <td>97.845090</td>\n",
       "      <td>0.056807</td>\n",
       "      <td>0.0091</td>\n",
       "      <td>0.0091</td>\n",
       "      <td>...</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7672</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248893146</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.015502</td>\n",
       "      <td>78.590880</td>\n",
       "      <td>0.031837</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>9_Het</td>\n",
       "      <td>14_ReprPCWk</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7673 rows Ã— 97 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr_x        pos ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0     chr1      77502     A     G     0.000761         0.014884   \n",
       "1     chr1      86065     G     C     0.069599         0.014811   \n",
       "2     chr1     883806     T     C     0.001913         0.015814   \n",
       "3     chr1     894725     T     C     0.000261         0.016190   \n",
       "4     chr1     988951     T     C     0.000174         0.014875   \n",
       "...    ...        ...   ...   ...          ...              ...   \n",
       "7668  chr1  248877982     A     C     0.002087         0.020471   \n",
       "7669  chr1  248879155     T     C     0.002696         0.015206   \n",
       "7670  chr1  248885752     T     C     0.000696         0.018184   \n",
       "7671  chr1  248888164     A     G     0.002261         0.015049   \n",
       "7672  chr1  248893146     A     C     0.000261         0.015502   \n",
       "\n",
       "      log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                  66.712830        0.027054             0.0002   \n",
       "1                 189.841190        0.106602             0.0869   \n",
       "2                  86.104660        0.048685             0.0006   \n",
       "3                 104.184900        0.052455             0.0001   \n",
       "4                  71.228340        0.038546           2.67e-05   \n",
       "...                      ...             ...                ...   \n",
       "7668              109.172440        0.046729             0.0035   \n",
       "7669               61.615257        0.028550             0.0285   \n",
       "7670              145.333740        0.086757             0.0002   \n",
       "7671               97.845090        0.056807             0.0091   \n",
       "7672               78.590880        0.031837                  .   \n",
       "\n",
       "     gnomad40_genome_AF_raw  ... Duodenum_Mucosa CD3_Primary_Cells_Cord_BI  \\\n",
       "0                    0.0002  ...        15_Quies                  15_Quies   \n",
       "1                    0.0892  ...        15_Quies                  15_Quies   \n",
       "2                    0.0007  ...            4_Tx                      4_Tx   \n",
       "3                    0.0001  ...          1_TssA                    1_TssA   \n",
       "4                    0.0004  ...          5_TxWk                  15_Quies   \n",
       "...                     ...  ...             ...                       ...   \n",
       "7668                 0.0035  ...           9_Het                     9_Het   \n",
       "7669                 0.0285  ...        15_Quies                     9_Het   \n",
       "7670                 0.0002  ...        15_Quies                  15_Quies   \n",
       "7671                 0.0091  ...        15_Quies                  15_Quies   \n",
       "7672                      .  ...        15_Quies                  15_Quies   \n",
       "\n",
       "     ES-I3_Cell_Line Psoas_Muscle Brain_Germinal_Matrix  \\\n",
       "0           15_Quies     15_Quies              15_Quies   \n",
       "1           15_Quies     15_Quies           14_ReprPCWk   \n",
       "2             5_TxWk       5_TxWk                  4_Tx   \n",
       "3          13_ReprPC       1_TssA             10_TssBiv   \n",
       "4             5_TxWk     15_Quies                  4_Tx   \n",
       "...              ...          ...                   ...   \n",
       "7668           9_Het        9_Het                 9_Het   \n",
       "7669           9_Het        9_Het              15_Quies   \n",
       "7670        15_Quies     15_Quies              15_Quies   \n",
       "7671        15_Quies     15_Quies              15_Quies   \n",
       "7672        15_Quies     15_Quies                 9_Het   \n",
       "\n",
       "     HeLa-S3_Cervical_Carcinoma        Lung Esophagus  \\\n",
       "0                      15_Quies    15_Quies  15_Quies   \n",
       "1                   14_ReprPCWk    15_Quies  15_Quies   \n",
       "2                        5_TxWk  8_ZNF/Rpts      4_Tx   \n",
       "3                        1_TssA      1_TssA    1_TssA   \n",
       "4                        5_TxWk        4_Tx      4_Tx   \n",
       "...                         ...         ...       ...   \n",
       "7668                   15_Quies    15_Quies  15_Quies   \n",
       "7669                   15_Quies    15_Quies  15_Quies   \n",
       "7670                   15_Quies    15_Quies  15_Quies   \n",
       "7671                   15_Quies    15_Quies  15_Quies   \n",
       "7672                14_ReprPCWk    15_Quies  15_Quies   \n",
       "\n",
       "     Chondrocytes_from_Bone_Marrow_Derived_Mesenchymal_Stem_Cell_Cultured_Cells  \\\n",
       "0                                              15_Quies                           \n",
       "1                                           14_ReprPCWk                           \n",
       "2                                                  4_Tx                           \n",
       "3                                                1_TssA                           \n",
       "4                                                  4_Tx                           \n",
       "...                                                 ...                           \n",
       "7668                                              9_Het                           \n",
       "7669                                           15_Quies                           \n",
       "7670                                              9_Het                           \n",
       "7671                                           15_Quies                           \n",
       "7672                                           15_Quies                           \n",
       "\n",
       "     Rectal_Mucosa.Donor_31  \n",
       "0                  15_Quies  \n",
       "1                  15_Quies  \n",
       "2                      4_Tx  \n",
       "3                    1_TssA  \n",
       "4                      4_Tx  \n",
       "...                     ...  \n",
       "7668               15_Quies  \n",
       "7669               15_Quies  \n",
       "7670               15_Quies  \n",
       "7671               15_Quies  \n",
       "7672               15_Quies  \n",
       "\n",
       "[7673 rows x 97 columns]"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [],
   "source": [
    "# snp_data=snp_data[['chr_x','start','end_x','anno_x','anno_y']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "name=\"tiss1\"\n",
    "name2=\"tiss2\"\n",
    "snp_data=snp_data.rename(columns={\"chr_x\":\"chr\",\"start\":\"start\",\"end_x\":\"end\",\"anno_x\":name,\"anno_y\":name2\n",
    "})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_x</th>\n",
       "      <th>alt_x</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>log_counts_diff</th>\n",
       "      <th>log_probs_diff_abs_sum</th>\n",
       "      <th>probs_jsd_diff</th>\n",
       "      <th>gnomad40_genome_AF</th>\n",
       "      <th>gnomad40_genome_AF_raw</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad40_genome_AF_ami</th>\n",
       "      <th>gnomad40_genome_AF_amr</th>\n",
       "      <th>gnomad40_genome_AF_asj</th>\n",
       "      <th>gnomad40_genome_AF_eas</th>\n",
       "      <th>gnomad40_genome_AF_fin</th>\n",
       "      <th>gnomad40_genome_AF_mid</th>\n",
       "      <th>gnomad40_genome_AF_nfe</th>\n",
       "      <th>gnomad40_genome_AF_remaining</th>\n",
       "      <th>gnomad40_genome_AF_sas</th>\n",
       "      <th>anno</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>77502</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000761</td>\n",
       "      <td>0.014884</td>\n",
       "      <td>66.712830</td>\n",
       "      <td>0.027054</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0035</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>86065</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.069599</td>\n",
       "      <td>0.014811</td>\n",
       "      <td>189.841190</td>\n",
       "      <td>0.106602</td>\n",
       "      <td>0.0869</td>\n",
       "      <td>0.0892</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1634</td>\n",
       "      <td>0.0369</td>\n",
       "      <td>0.1826</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>0.1600</td>\n",
       "      <td>0.0459</td>\n",
       "      <td>0.1497</td>\n",
       "      <td>0.0739</td>\n",
       "      <td>0.0775</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>883806</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001913</td>\n",
       "      <td>0.015814</td>\n",
       "      <td>86.104660</td>\n",
       "      <td>0.048685</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0177</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.47e-05</td>\n",
       "      <td>0.0014</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>4_Tx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>894725</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.016190</td>\n",
       "      <td>104.184900</td>\n",
       "      <td>0.052455</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10_TssBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>988951</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.014875</td>\n",
       "      <td>71.228340</td>\n",
       "      <td>0.038546</td>\n",
       "      <td>2.67e-05</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.482e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4_Tx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6826</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248877982</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002087</td>\n",
       "      <td>0.020471</td>\n",
       "      <td>109.172440</td>\n",
       "      <td>0.046729</td>\n",
       "      <td>0.0035</td>\n",
       "      <td>0.0035</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0021</td>\n",
       "      <td>0.0029</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0025</td>\n",
       "      <td>0.0068</td>\n",
       "      <td>0.0059</td>\n",
       "      <td>0.0024</td>\n",
       "      <td>0.0031</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6827</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248879155</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002696</td>\n",
       "      <td>0.015206</td>\n",
       "      <td>61.615257</td>\n",
       "      <td>0.028550</td>\n",
       "      <td>0.0285</td>\n",
       "      <td>0.0285</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0356</td>\n",
       "      <td>0.0248</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0480</td>\n",
       "      <td>0.0306</td>\n",
       "      <td>0.0400</td>\n",
       "      <td>0.0398</td>\n",
       "      <td>0.0054</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6828</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248885752</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000696</td>\n",
       "      <td>0.018184</td>\n",
       "      <td>145.333740</td>\n",
       "      <td>0.086757</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>6.531e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0048</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.94e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>9_Het</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6829</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248888164</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.002261</td>\n",
       "      <td>0.015049</td>\n",
       "      <td>97.845090</td>\n",
       "      <td>0.056807</td>\n",
       "      <td>0.0091</td>\n",
       "      <td>0.0091</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0067</td>\n",
       "      <td>0.0078</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0068</td>\n",
       "      <td>0.0152</td>\n",
       "      <td>0.0047</td>\n",
       "      <td>0.0039</td>\n",
       "      <td>15_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6830</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248893146</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>0.015502</td>\n",
       "      <td>78.590880</td>\n",
       "      <td>0.031837</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>9_Het</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6831 rows Ã— 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr        pos ref_x alt_x  allele_freq  log_counts_diff  \\\n",
       "0     chr1      77502     A     G     0.000761         0.014884   \n",
       "1     chr1      86065     G     C     0.069599         0.014811   \n",
       "2     chr1     883806     T     C     0.001913         0.015814   \n",
       "3     chr1     894725     T     C     0.000261         0.016190   \n",
       "4     chr1     988951     T     C     0.000174         0.014875   \n",
       "...    ...        ...   ...   ...          ...              ...   \n",
       "6826  chr1  248877982     A     C     0.002087         0.020471   \n",
       "6827  chr1  248879155     T     C     0.002696         0.015206   \n",
       "6828  chr1  248885752     T     C     0.000696         0.018184   \n",
       "6829  chr1  248888164     A     G     0.002261         0.015049   \n",
       "6830  chr1  248893146     A     C     0.000261         0.015502   \n",
       "\n",
       "      log_probs_diff_abs_sum  probs_jsd_diff gnomad40_genome_AF  \\\n",
       "0                  66.712830        0.027054             0.0002   \n",
       "1                 189.841190        0.106602             0.0869   \n",
       "2                  86.104660        0.048685             0.0006   \n",
       "3                 104.184900        0.052455             0.0001   \n",
       "4                  71.228340        0.038546           2.67e-05   \n",
       "...                      ...             ...                ...   \n",
       "6826              109.172440        0.046729             0.0035   \n",
       "6827               61.615257        0.028550             0.0285   \n",
       "6828              145.333740        0.086757             0.0002   \n",
       "6829               97.845090        0.056807             0.0091   \n",
       "6830               78.590880        0.031837                  .   \n",
       "\n",
       "     gnomad40_genome_AF_raw  ... gnomad40_genome_AF_ami  \\\n",
       "0                    0.0002  ...                      0   \n",
       "1                    0.0892  ...                 0.1634   \n",
       "2                    0.0007  ...                      0   \n",
       "3                    0.0001  ...                      0   \n",
       "4                    0.0004  ...                      0   \n",
       "...                     ...  ...                    ...   \n",
       "6826                 0.0035  ...                      0   \n",
       "6827                 0.0285  ...                 0.0066   \n",
       "6828                 0.0002  ...                      0   \n",
       "6829                 0.0091  ...                      0   \n",
       "6830                      .  ...                      .   \n",
       "\n",
       "     gnomad40_genome_AF_amr gnomad40_genome_AF_asj gnomad40_genome_AF_eas  \\\n",
       "0                         0                      0                 0.0035   \n",
       "1                    0.0369                 0.1826                 0.0004   \n",
       "2                    0.0001                      0                 0.0177   \n",
       "3                         0                      0                 0.0002   \n",
       "4                         0                      0                      0   \n",
       "...                     ...                    ...                    ...   \n",
       "6826                 0.0021                 0.0029                      0   \n",
       "6827                 0.0356                 0.0248                      0   \n",
       "6828              6.531e-05                      0                 0.0048   \n",
       "6829                 0.0067                 0.0078                      0   \n",
       "6830                      .                      .                      .   \n",
       "\n",
       "     gnomad40_genome_AF_fin gnomad40_genome_AF_mid gnomad40_genome_AF_nfe  \\\n",
       "0                         0                      0                      0   \n",
       "1                    0.1600                 0.0459                 0.1497   \n",
       "2                         0                      0               1.47e-05   \n",
       "3                         0                      0                 0.0003   \n",
       "4                         0                      0              1.482e-05   \n",
       "...                     ...                    ...                    ...   \n",
       "6826                 0.0025                 0.0068                 0.0059   \n",
       "6827                 0.0480                 0.0306                 0.0400   \n",
       "6828                      0                      0               2.94e-05   \n",
       "6829                 0.0066                 0.0068                 0.0152   \n",
       "6830                      .                      .                      .   \n",
       "\n",
       "     gnomad40_genome_AF_remaining gnomad40_genome_AF_sas       anno  \n",
       "0                               0                 0.0006   15_Quies  \n",
       "1                          0.0739                 0.0775   15_Quies  \n",
       "2                          0.0014                 0.0002       4_Tx  \n",
       "3                               0                      0  10_TssBiv  \n",
       "4                               0                      0       4_Tx  \n",
       "...                           ...                    ...        ...  \n",
       "6826                       0.0024                 0.0031   15_Quies  \n",
       "6827                       0.0398                 0.0054   15_Quies  \n",
       "6828                            0                 0.0002      9_Het  \n",
       "6829                       0.0047                 0.0039   15_Quies  \n",
       "6830                            .                      .      9_Het  \n",
       "\n",
       "[6831 rows x 28 columns]"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
